{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-25T15:03:19.725109Z",
     "start_time": "2020-11-25T15:03:19.575Z"
    }
   },
   "outputs": [],
   "source": [
    "#installation, may not be needed\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "\n",
    "#BiocManager::install(\"DESeq2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-25T15:03:24.548458Z",
     "start_time": "2020-11-25T15:03:19.577Z"
    }
   },
   "outputs": [],
   "source": [
    "suppressMessages(library(DESeq2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-25T15:04:38.558807Z",
     "start_time": "2020-11-25T15:03:19.579Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#UPF1 knockdown vs scramble\n",
    "count_file <- \"colombo_combined_counts.tsv\"\n",
    "row_names <- \"tx_id\"\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('scr1_Homo_sapiens_RNA.Seq', 'scr2_Homo_sapiens_RNA.Seq', 'scr3_Homo_sapiens_RNA.Seq', 'UPF1_KD1_Homo_sapiens_RNA.Seq', 'UPF1_KD2_Homo_sapiens_RNA.Seq', 'UPF1_KD3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/UPF1_kd_vs_scr.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#UPF1 rescue vs knockdown\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('UPF1_KD1_Homo_sapiens_RNA.Seq', 'UPF1_KD2_Homo_sapiens_RNA.Seq', 'UPF1_KD3_Homo_sapiens_RNA.Seq', 'UPF1_rescue1_Homo_sapiens_RNA.Seq', 'UPF1_rescue2_Homo_sapiens_RNA.Seq', 'UPF1_rescue3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/UPF1_rescue_vs_kd.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG6 knockdown vs scramble\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('scr1_Homo_sapiens_RNA.Seq', 'scr2_Homo_sapiens_RNA.Seq', 'scr3_Homo_sapiens_RNA.Seq', 'SMG6_KD1_Homo_sapiens_RNA.Seq', 'SMG6_KD2_Homo_sapiens_RNA.Seq', 'SMG6_KD3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/SMG6_kd_vs_scr.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG6 rescue vs knockdown\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('SMG6_KD1_Homo_sapiens_RNA.Seq', 'SMG6_KD2_Homo_sapiens_RNA.Seq', 'SMG6_KD3_Homo_sapiens_RNA.Seq', 'SMG6_rescue1_Homo_sapiens_RNA.Seq', 'SMG6_rescue2_Homo_sapiens_RNA.Seq', 'SMG6_rescue3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/SMG6_rescue_vs_kd.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG7 knockdown vs scramble\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('scr4_Homo_sapiens_RNA.Seq', 'scr5_Homo_sapiens_RNA.Seq', 'scr6_Homo_sapiens_RNA.Seq', 'SMG7_KD1_Homo_sapiens_RNA.Seq', 'SMG7_KD2_Homo_sapiens_RNA.Seq', 'SMG7_KD3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/SMG7_kd_vs_scr.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG7 rescue vs knockdown\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('SMG7_KD1_Homo_sapiens_RNA.Seq', 'SMG7_KD2_Homo_sapiens_RNA.Seq', 'SMG7_KD3_Homo_sapiens_RNA.Seq', 'SMG7_rescue1_Homo_sapiens_RNA.Seq', 'SMG7_rescue2_Homo_sapiens_RNA.Seq', 'SMG7_rescue3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/SMG7_rescue_vs_kd.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG6/7 double_knockdown vs scramble\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('scr4_Homo_sapiens_RNA.Seq', 'scr5_Homo_sapiens_RNA.Seq', 'scr6_Homo_sapiens_RNA.Seq', 'dKD1_Homo_sapiens_RNA.Seq', 'dKD2_Homo_sapiens_RNA.Seq', 'dKD3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/dSMG_kd_vs_scr.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG6 rescue vs SMG6/7 double knockdown\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('dKD1_Homo_sapiens_RNA.Seq', 'dKD2_Homo_sapiens_RNA.Seq', 'dKD3_Homo_sapiens_RNA.Seq', 'dKD_SMG6_rescue1_Homo_sapiens_RNA.Seq', 'dKD_SMG6_rescue2_Homo_sapiens_RNA.Seq', 'dKD_SMG6_rescue3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/SMG6_rescue_vs_dSMG_kd.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#SMG7 rescue vs SMG6/7 double knockdown\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('dKD1_Homo_sapiens_RNA.Seq', 'dKD2_Homo_sapiens_RNA.Seq', 'dKD3_Homo_sapiens_RNA.Seq', 'dKD_SMG7_rescue1_Homo_sapiens_RNA.Seq', 'dKD_SMG7_rescue2_Homo_sapiens_RNA.Seq', 'dKD_SMG7_rescue3_Homo_sapiens_RNA.Seq')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 3), rep(\"exp\", 3)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/SMG7_rescue_vs_dSMG_kd.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
