{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download datasets from encode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!curl -L https://www.encodeproject.org/files/ENCFF954MLV/@@download/ENCFF954MLV.fastq.gz -o HepG2_EIF4A3_1_r1.fastq.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First I need to assemble the list of FASTQ files from the pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:28:15.065902Z",
     "start_time": "2020-10-20T02:28:14.385914Z"
    }
   },
   "outputs": [],
   "source": [
    "#add pipeline to path\n",
    "import sys\n",
    "import datetime\n",
    "\n",
    "\n",
    "pipeline_dir = '/home/boris/analysis_pipelines/ribo_seq/'\n",
    "if pipeline_dir not in sys.path:\n",
    "    sys.path.append(pipeline_dir)\n",
    "\n",
    "import ribo_seq_main\n",
    "import ribo_settings\n",
    "import ribo_utils\n",
    "import ribo_lib\n",
    "import ribo_qc\n",
    "import ribo_plotting\n",
    "import ribo_tables\n",
    "\n",
    "def current_time():\n",
    "    return datetime.datetime.now().strftime(\"%Y%m%d_%H%M\")\n",
    "def current_date():\n",
    "    return datetime.datetime.now().strftime(\"%Y%m%d\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:28:16.988783Z",
     "start_time": "2020-10-20T02:28:16.953920Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "created data-free shell for WT_DMSO_1_mono\n",
      "created data-free shell for WT_DMSO_2_mono\n",
      "created data-free shell for WT_SMG1i_1_mono\n",
      "created data-free shell for WT_SMG1i_2_mono\n",
      "created data-free shell for eIF4E2_4_DMSO_1_mono\n",
      "created data-free shell for eIF4E2_4_DMSO_2_mono\n",
      "created data-free shell for eIF4E2_4_SMG1i_1_mono\n",
      "created data-free shell for eIF4E2_4_SMG1i_2_mono\n",
      "created data-free shell for GIGYF2_4_DMSO_1_mono\n",
      "created data-free shell for GIGYF2_4_DMSO_2_mono\n",
      "created data-free shell for GIGYF2_4_SMG1i_1_mono\n",
      "created data-free shell for GIGYF2_4_SMG1i_2_mono\n",
      "created data-free shell for WT_DMSO_1_di\n",
      "created data-free shell for WT_DMSO_2_di\n",
      "created data-free shell for WT_SMG1i_1_di\n",
      "created data-free shell for WT_SMG1i_2_di\n",
      "created data-free shell for eIF4E2_4_DMSO_1_di\n",
      "created data-free shell for eIF4E2_4_DMSO_2_di\n",
      "created data-free shell for eIF4E2_4_SMG1i_1_di\n",
      "created data-free shell for eIF4E2_4_SMG1i_2_di\n",
      "created data-free shell for GIGYF2_4_DMSO_1_di\n",
      "created data-free shell for GIGYF2_4_DMSO_2_di\n",
      "created data-free shell for GIGYF2_4_SMG1i_1_di\n",
      "created data-free shell for GIGYF2_4_SMG1i_2_di\n",
      "created data-free shell for WT_DMSO_1_RNA\n",
      "created data-free shell for WT_DMSO_2_RNA\n",
      "created data-free shell for WT_SMG1i_1_RNA\n",
      "created data-free shell for WT_SMG1i_2_RNA\n",
      "created data-free shell for eIF4E2_4_DMSO_1_RNA\n",
      "created data-free shell for eIF4E2_4_DMSO_2_RNA\n",
      "created data-free shell for eIF4E2_4_SMG1i_1_RNA\n",
      "created data-free shell for eIF4E2_4_SMG1i_2_RNA\n",
      "created data-free shell for GIGYF2_4_DMSO_1_RNA\n",
      "created data-free shell for GIGYF2_4_DMSO_2_RNA\n",
      "created data-free shell for GIGYF2_4_SMG1i_1_RNA\n",
      "created data-free shell for GIGYF2_4_SMG1i_2_RNA\n",
      "created data-free shell for NT1_1_mono\n",
      "created data-free shell for NT1_2_mono\n",
      "created data-free shell for SMG1i_1_mono\n",
      "created data-free shell for SMG1i_2_mono\n",
      "created data-free shell for eIF4E2_1_mono\n",
      "created data-free shell for eIF4E2_2_mono\n",
      "created data-free shell for GIGYF2_1_mono\n",
      "created data-free shell for GIGYF2_2_mono\n",
      "created data-free shell for NT1_1_RNA\n",
      "created data-free shell for NT1_2_RNA\n",
      "created data-free shell for SMG1i_1_RNA\n",
      "created data-free shell for SMG1i_2_RNA\n",
      "created data-free shell for eIF4E2_1_RNA\n",
      "created data-free shell for eIF4E2_2_RNA\n",
      "created data-free shell for GIGYF2_1_RNA\n",
      "created data-free shell for GIGYF2_2_RNA\n"
     ]
    }
   ],
   "source": [
    "## Now import the pickled read count library objects\n",
    "import os\n",
    "import os.path as path\n",
    "\n",
    "threads = 48\n",
    "\n",
    "mono_settings_file = '../8.019_mono_settings_genome.json.txt'\n",
    "mono_settings = ribo_settings.ribo_settings(mono_settings_file)\n",
    "mono_experiment = ribo_seq_main.experiment(mono_settings, threads, process_data=False)\n",
    "\n",
    "di_settings_file = '../8.019_di_settings_genome.json.txt'\n",
    "di_settings = ribo_settings.ribo_settings(di_settings_file)\n",
    "di_experiment = ribo_seq_main.experiment(di_settings, threads, process_data=False)\n",
    "\n",
    "RNA_settings_file = '../8.019_RNA_settings_genome.json.txt'\n",
    "RNA_settings = ribo_settings.ribo_settings(RNA_settings_file)\n",
    "RNA_experiment = ribo_seq_main.experiment(RNA_settings, threads, process_data=False)\n",
    "\n",
    "old_mono_settings_file = '../../5.049_readthrough_analysis/RPF_SMG1i_GIGY_4E2.5.049.json.txt'\n",
    "old_mono_settings = ribo_settings.ribo_settings(old_mono_settings_file)\n",
    "old_mono_experiment = ribo_seq_main.experiment(old_mono_settings, threads, process_data=False)\n",
    "\n",
    "old_RNA_settings_file = '../../5.049_readthrough_analysis/RNA_SMG1i_GIGY_4E2.5.049.json.txt'\n",
    "old_RNA_settings = ribo_settings.ribo_settings(old_RNA_settings_file)\n",
    "old_RNA_experiment = ribo_seq_main.experiment(old_RNA_settings, threads, process_data=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:28:42.404295Z",
     "start_time": "2020-10-20T02:28:42.381202Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/WT_DMSO_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/WT_DMSO_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/WT_DMSO_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/WT_DMSO_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/WT_SMG1i_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/WT_SMG1i_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/WT_SMG1i_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/WT_SMG1i_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_DMSO_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_DMSO_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_SMG1i_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_SMG1i_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_DMSO_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_DMSO_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_SMG1i_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_mono_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_SMG1i_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/WT_DMSO_1_diUnmapped.out.mate1) --validateMappings -o salmon_results/WT_DMSO_1_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/WT_DMSO_2_diUnmapped.out.mate1) --validateMappings -o salmon_results/WT_DMSO_2_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/WT_SMG1i_1_diUnmapped.out.mate1) --validateMappings -o salmon_results/WT_SMG1i_1_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/WT_SMG1i_2_diUnmapped.out.mate1) --validateMappings -o salmon_results/WT_SMG1i_2_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_1_diUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_DMSO_1_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_2_diUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_DMSO_2_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_1_diUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_SMG1i_1_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_2_diUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_4_SMG1i_2_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_1_diUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_DMSO_1_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_2_diUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_DMSO_2_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_1_diUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_SMG1i_1_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_di_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_2_diUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_4_SMG1i_2_di\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/NT1_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/NT1_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/NT1_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/NT1_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/SMG1i_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/SMG1i_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/SMG1i_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/SMG1i_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/eIF4E2_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/eIF4E2_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_2_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/GIGYF2_1_monoUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_1_mono\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/ -l SF -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RPF_pipeline/ncrna_mapped_reads/GIGYF2_2_monoUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_2_mono\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "logfile = open('20200823_salmon_run.log', 'w')\n",
    "index_folder = '../20200719_make_salmon_indices/indices/superset_stops_collapsed_CDS_15_15_k13/'\n",
    "for lib in mono_experiment.libs + di_experiment.libs+old_mono_experiment.libs:\n",
    "    fastq_file = lib.lib_settings.get_ncrna_unmapped_reads()\n",
    "    out_folder = os.path.join('salmon_results', lib.lib_settings.sample_name)\n",
    "    salmon_cmd=\"salmon quant -i {index_folder} -l SF -p 40 -r <(cat {fastq_file}) --validateMappings -o {out_folder}\".format(**locals())\n",
    "    logfile.write('%s\\n' % salmon_cmd)\n",
    "    print(salmon_cmd)\n",
    "    #subprocess.Popen(salmon_cmd, shell=True).wait()\n",
    "    #s = subprocess.Popen(salmon_cmd, shell=False)\n",
    "    #out, err = s.communicate()\n",
    "    #logfile.write(out)\n",
    "    #logfile.write(err)\n",
    "#logfile.close()\n",
    "#something is wrong, so just pasted the commands to the command line and it worked"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:28:42.998264Z",
     "start_time": "2020-10-20T02:28:42.982768Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_DMSO_1_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_DMSO_1_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/WT_DMSO_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_DMSO_2_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_DMSO_2_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/WT_DMSO_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_SMG1i_1_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_SMG1i_1_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/WT_SMG1i_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_SMG1i_2_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/WT_SMG1i_2_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/WT_SMG1i_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_1_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_1_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/eIF4E2_4_DMSO_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_2_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_DMSO_2_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/eIF4E2_4_DMSO_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_1_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_1_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/eIF4E2_4_SMG1i_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_2_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/eIF4E2_4_SMG1i_2_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/eIF4E2_4_SMG1i_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_1_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_1_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/GIGYF2_4_DMSO_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_2_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_DMSO_2_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/GIGYF2_4_DMSO_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_1_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_1_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/GIGYF2_4_SMG1i_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l ISR -p 40 -1 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_2_RNAUnmapped.out.mate1) -2 <(cat /fastscratch/boris/8.019_profiling_analysis/pipeline_RNA_genome/ncrna_mapped_reads/GIGYF2_4_SMG1i_2_RNAUnmapped.out.mate2) --gcBias --seqBias --validateMappings -o salmon_results/GIGYF2_4_SMG1i_2_RNA\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "#logfile = open('20200720_salmon_run_RNA.log', 'w')\n",
    "index_folder = '../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/'\n",
    "for lib in RNA_experiment.libs:\n",
    "    read1_file, read2_file = lib.lib_settings.get_ncrna_unmapped_reads()\n",
    "    out_folder = os.path.join('salmon_results', lib.lib_settings.sample_name)\n",
    "    salmon_cmd=\"salmon quant -i {index_folder} -l ISR -p 40 -1 <(cat {read1_file}) -2 <(cat {read2_file}) --gcBias --seqBias --validateMappings -o {out_folder}\".format(**locals())\n",
    "    logfile.write('%s\\n' % salmon_cmd)\n",
    "    print(salmon_cmd)\n",
    "    #subprocess.Popen(salmon_cmd, shell=True).wait()\n",
    "    #s = subprocess.Popen(salmon_cmd, shell=False)\n",
    "    #out, err = s.communicate()\n",
    "    #logfile.write(out)\n",
    "    #logfile.write(err)\n",
    "#logfile.close()\n",
    "#something is wrong, so just pasted the commands to the command line and it worked"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:28:43.572532Z",
     "start_time": "2020-10-20T02:28:43.556440Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/NT1_1_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/NT1_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/NT1_2_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/NT1_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/SMG1i_1_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/SMG1i_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/SMG1i_2_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/SMG1i_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/eIF4E2_1_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/eIF4E2_2_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/eIF4E2_2_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/GIGYF2_1_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_1_RNA\n",
      "salmon quant -i ../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/ -l SR -p 40 -r <(cat /fastscratch/boris/5.049_readthrough_analysis/SMG1i_GIGYF2_EIF4E2_RNA_pipeline/ncrna_mapped_reads/GIGYF2_2_RNAUnmapped.out.mate1) --validateMappings -o salmon_results/GIGYF2_2_RNA\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "#logfile = open('20200720_salmon_run.log', 'w')\n",
    "index_folder = '../20200719_make_salmon_indices/indices/superset_stops_collapsed_tx_k31/'\n",
    "for lib in old_RNA_experiment.libs:\n",
    "    fastq_file = lib.lib_settings.get_ncrna_unmapped_reads()\n",
    "    out_folder = os.path.join('salmon_results', lib.lib_settings.sample_name)\n",
    "    salmon_cmd=\"salmon quant -i {index_folder} -l SR -p 40 -r <(cat {fastq_file}) --validateMappings -o {out_folder}\".format(**locals())\n",
    "    logfile.write('%s\\n' % salmon_cmd)\n",
    "    print(salmon_cmd)\n",
    "    #subprocess.Popen(salmon_cmd, shell=True).wait()\n",
    "    #s = subprocess.Popen(salmon_cmd, shell=False)\n",
    "    #out, err = s.communicate()\n",
    "    #logfile.write(out)\n",
    "    #logfile.write(err)\n",
    "logfile.close()\n",
    "#something is wrong, so just pasted the commands to the command line and it worked"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now compile the counts from Salmon output files into a single dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:29:03.152188Z",
     "start_time": "2020-10-20T02:28:48.180583Z"
    }
   },
   "outputs": [],
   "source": [
    "#code to add gene names and ids to files\n",
    "comprehensive_gtf_data = ribo_utils.gtf_data('../20200715_orfquant/orfquant_tx_sets/superset_stops_collapsed.gtf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:29:03.169034Z",
     "start_time": "2020-10-20T02:29:03.156608Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_nmd_status(x):\n",
    "    if '|' in x:\n",
    "        return get_nmd_status(x.split('|')[0])\n",
    "    return sorted(comprehensive_gtf_data.transcript_to_entries[x])[0].get_value('transcript_type') == 'nonsense_mediated_decay'\n",
    "\n",
    "def get_gene_id(x):\n",
    "    if '|' in x:\n",
    "        return get_gene_id(x.split('|')[0])\n",
    "    return comprehensive_gtf_data.tx_to_genes[x]\n",
    "\n",
    "def get_gene_name(x):\n",
    "    if '|' in x:\n",
    "        return get_gene_name(x.split('|')[0])\n",
    "    return comprehensive_gtf_data.tx_to_gene_names[x]\n",
    "\n",
    "def log_trans(x):\n",
    "    try:\n",
    "        return math.log(x, 10)\n",
    "    except:\n",
    "        return float('NaN')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:29:28.869366Z",
     "start_time": "2020-10-20T02:29:03.172222Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "dfs = []\n",
    "\n",
    "for lib in mono_experiment.libs + di_experiment.libs+RNA_experiment.libs+old_mono_experiment.libs+old_RNA_experiment.libs:\n",
    "    expression_file = os.path.join('salmon_results', lib.lib_settings.sample_name, 'quant.sf')\n",
    "    df = pd.read_csv(expression_file, sep='\\t')\n",
    "    df = df[['Name', 'NumReads']]\n",
    "    df.rename(columns = {'Name':'tx_id', 'NumReads':lib.lib_settings.sample_name}, inplace = True)\n",
    "    dfs.append(df)\n",
    "combined_read_counts_df = reduce(lambda x, y: pd.merge(x, y, on = 'tx_id', how='outer'), dfs)\n",
    "combined_read_counts_df['gene_id'] = combined_read_counts_df['tx_id'].apply(get_gene_id)\n",
    "combined_read_counts_df['gene_name'] = combined_read_counts_df['tx_id'].apply(get_gene_name)\n",
    "combined_read_counts_df.dropna().to_csv('20200823_all_datasets_stop_collapsed_counts.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:29:54.231654Z",
     "start_time": "2020-10-20T02:29:28.873197Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "dfs = []\n",
    "    \n",
    "for lib in mono_experiment.libs + di_experiment.libs+RNA_experiment.libs+old_mono_experiment.libs+old_RNA_experiment.libs:\n",
    "    expression_file = os.path.join('salmon_results', lib.lib_settings.sample_name, 'quant.sf')\n",
    "    df = pd.read_csv(expression_file, sep='\\t')\n",
    "    df = df[['Name', 'TPM']]\n",
    "    df.rename(columns = {'Name':'tx_id', 'TPM':lib.lib_settings.sample_name}, inplace = True)\n",
    "    dfs.append(df)\n",
    "    \n",
    "combined_read_TPM_df = reduce(lambda x, y: pd.merge(x, y, on = 'tx_id', how='outer'), dfs)\n",
    "combined_read_TPM_df['gene_id'] = combined_read_TPM_df['tx_id'].apply(get_gene_id)\n",
    "combined_read_TPM_df['gene_name'] = combined_read_TPM_df['tx_id'].apply(get_gene_name)\n",
    "combined_read_TPM_df.dropna().to_csv('20200823_all_datasets_stop_collapsed_tpm.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:29:54.252049Z",
     "start_time": "2020-10-20T02:29:54.236527Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'WT_DMSO_1_mono', u'WT_DMSO_2_mono', u'WT_SMG1i_1_mono',\n",
       "       u'WT_SMG1i_2_mono', u'eIF4E2_4_DMSO_1_mono', u'eIF4E2_4_DMSO_2_mono',\n",
       "       u'eIF4E2_4_SMG1i_1_mono', u'eIF4E2_4_SMG1i_2_mono',\n",
       "       u'GIGYF2_4_DMSO_1_mono', u'GIGYF2_4_DMSO_2_mono',\n",
       "       u'GIGYF2_4_SMG1i_1_mono', u'GIGYF2_4_SMG1i_2_mono', u'WT_DMSO_1_di',\n",
       "       u'WT_DMSO_2_di', u'WT_SMG1i_1_di', u'WT_SMG1i_2_di',\n",
       "       u'eIF4E2_4_DMSO_1_di', u'eIF4E2_4_DMSO_2_di', u'eIF4E2_4_SMG1i_1_di',\n",
       "       u'eIF4E2_4_SMG1i_2_di', u'GIGYF2_4_DMSO_1_di', u'GIGYF2_4_DMSO_2_di',\n",
       "       u'GIGYF2_4_SMG1i_1_di', u'GIGYF2_4_SMG1i_2_di', u'WT_DMSO_1_RNA',\n",
       "       u'WT_DMSO_2_RNA', u'WT_SMG1i_1_RNA', u'WT_SMG1i_2_RNA',\n",
       "       u'eIF4E2_4_DMSO_1_RNA', u'eIF4E2_4_DMSO_2_RNA', u'eIF4E2_4_SMG1i_1_RNA',\n",
       "       u'eIF4E2_4_SMG1i_2_RNA', u'GIGYF2_4_DMSO_1_RNA', u'GIGYF2_4_DMSO_2_RNA',\n",
       "       u'GIGYF2_4_SMG1i_1_RNA', u'GIGYF2_4_SMG1i_2_RNA', u'NT1_1_mono',\n",
       "       u'NT1_2_mono', u'SMG1i_1_mono', u'SMG1i_2_mono', u'eIF4E2_1_mono',\n",
       "       u'eIF4E2_2_mono', u'GIGYF2_1_mono', u'GIGYF2_2_mono', u'NT1_1_RNA',\n",
       "       u'NT1_2_RNA', u'SMG1i_1_RNA', u'SMG1i_2_RNA', u'eIF4E2_1_RNA',\n",
       "       u'eIF4E2_2_RNA', u'GIGYF2_1_RNA', u'GIGYF2_2_RNA'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_read_counts_df.columns[1:-2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now sum counts for transcripts in 2 ways: combining all tx for a given gene, and all tx with the same genic stop codon."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:31:51.000765Z",
     "start_time": "2020-10-20T02:29:54.255713Z"
    }
   },
   "outputs": [],
   "source": [
    "##merging all counts and tpms for the same gene\n",
    "datasets = list(combined_read_counts_df.columns[1:-2])\n",
    "\n",
    "data = []\n",
    "grouped_counts = combined_read_counts_df.groupby('gene_id')\n",
    "for gene_id in combined_read_counts_df['gene_id'].unique():\n",
    "    gene_transcript_data = grouped_counts.get_group(gene_id)\n",
    "    tx_ids = '|'.join(sorted(gene_transcript_data['tx_id'].unique()))\n",
    "    data.append([tx_ids] + [gene_transcript_data[dataset].sum() for dataset in datasets])\n",
    "counts_by_gene_df = pd.DataFrame(data, columns = ['tx_ids']+datasets)\n",
    "#counts_by_gene_df['gene_id'] = counts_by_gene_df['tx_ids'].apply(get_gene_id)\n",
    "#counts_by_gene_df['gene_name'] = counts_by_gene_df['tx_ids'].apply(get_gene_name)\n",
    "counts_by_gene_df.to_csv('20200823_summed_gene_counts.tsv', sep='\\t', index = False)\n",
    "\n",
    "datasets = list(combined_read_TPM_df.columns[1:-2])\n",
    "data = []\n",
    "grouped_TPMs = combined_read_TPM_df.groupby('gene_id')\n",
    "for gene_id in combined_read_TPM_df['gene_id'].unique():\n",
    "    gene_transcript_data = grouped_TPMs.get_group(gene_id)\n",
    "    tx_ids = '|'.join(sorted(gene_transcript_data['tx_id'].unique()))\n",
    "    data.append([tx_ids] + [gene_transcript_data[dataset].sum() for dataset in datasets])\n",
    "tpm_by_gene_df = pd.DataFrame(data, columns = ['tx_ids']+datasets)\n",
    "tpm_by_gene_df['gene_id'] = tpm_by_gene_df['tx_ids'].apply(get_gene_id)\n",
    "tpm_by_gene_df['gene_name'] = tpm_by_gene_df['tx_ids'].apply(get_gene_name)\n",
    "tpm_by_gene_df.to_csv('20200823_summed_gene_TPMs.tsv', sep='\\t', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### now combining transcripts by stop codon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:31:51.013019Z",
     "start_time": "2020-10-20T02:31:51.005374Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_genomic_stop_coords(x):\n",
    "    #if given a |-seperated list of tx, will only return value for first one\n",
    "    if '|' in x:\n",
    "        return get_genomic_stop_coords(x.split('|')[0])\n",
    "    stop_entry = comprehensive_gtf_data.sorted_exons(x, exon_type='stop_codon')\n",
    "    if len(stop_entry) == 0:\n",
    "        return 'not annotated'\n",
    "    stop = '%s %s %s %s' % (stop_entry[0].get_value('chr'), stop_entry[0].get_value('strand'), \n",
    "                             stop_entry[0].get_value('start'), stop_entry[0].get_value('end'))\n",
    "    return stop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:31:52.639800Z",
     "start_time": "2020-10-20T02:31:51.016976Z"
    }
   },
   "outputs": [],
   "source": [
    "combined_read_counts_df['stop_codon'] = combined_read_counts_df['tx_id'].apply(get_genomic_stop_coords)\n",
    "combined_read_TPM_df['stop_codon'] = combined_read_TPM_df['tx_id'].apply(get_genomic_stop_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:31:54.154853Z",
     "start_time": "2020-10-20T02:31:52.644070Z"
    }
   },
   "outputs": [],
   "source": [
    "combined_read_counts_df.sum(axis=0).to_csv('20201005_total_readcounts.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:39.561885Z",
     "start_time": "2020-10-20T02:31:54.158835Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['WT_DMSO_1_mono', 'WT_DMSO_2_mono', 'WT_SMG1i_1_mono', 'WT_SMG1i_2_mono', 'eIF4E2_4_DMSO_1_mono', 'eIF4E2_4_DMSO_2_mono', 'eIF4E2_4_SMG1i_1_mono', 'eIF4E2_4_SMG1i_2_mono', 'GIGYF2_4_DMSO_1_mono', 'GIGYF2_4_DMSO_2_mono', 'GIGYF2_4_SMG1i_1_mono', 'GIGYF2_4_SMG1i_2_mono', 'WT_DMSO_1_di', 'WT_DMSO_2_di', 'WT_SMG1i_1_di', 'WT_SMG1i_2_di', 'eIF4E2_4_DMSO_1_di', 'eIF4E2_4_DMSO_2_di', 'eIF4E2_4_SMG1i_1_di', 'eIF4E2_4_SMG1i_2_di', 'GIGYF2_4_DMSO_1_di', 'GIGYF2_4_DMSO_2_di', 'GIGYF2_4_SMG1i_1_di', 'GIGYF2_4_SMG1i_2_di', 'WT_DMSO_1_RNA', 'WT_DMSO_2_RNA', 'WT_SMG1i_1_RNA', 'WT_SMG1i_2_RNA', 'eIF4E2_4_DMSO_1_RNA', 'eIF4E2_4_DMSO_2_RNA', 'eIF4E2_4_SMG1i_1_RNA', 'eIF4E2_4_SMG1i_2_RNA', 'GIGYF2_4_DMSO_1_RNA', 'GIGYF2_4_DMSO_2_RNA', 'GIGYF2_4_SMG1i_1_RNA', 'GIGYF2_4_SMG1i_2_RNA', 'NT1_1_mono', 'NT1_2_mono', 'SMG1i_1_mono', 'SMG1i_2_mono', 'eIF4E2_1_mono', 'eIF4E2_2_mono', 'GIGYF2_1_mono', 'GIGYF2_2_mono', 'NT1_1_RNA', 'NT1_2_RNA', 'SMG1i_1_RNA', 'SMG1i_2_RNA', 'eIF4E2_1_RNA', 'eIF4E2_2_RNA', 'GIGYF2_1_RNA', 'GIGYF2_2_RNA']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/boris/.local/lib/python2.7/site-packages/ipykernel_launcher.py:6: FutureWarning: Interpreting tuple 'by' as a list of keys, rather than a single key. Use 'by=[...]' instead of 'by=(...)'. In the future, a tuple will always mean a single key.\n",
      "  \n",
      "/home/boris/.local/lib/python2.7/site-packages/ipykernel_launcher.py:20: FutureWarning: Interpreting tuple 'by' as a list of keys, rather than a single key. Use 'by=[...]' instead of 'by=(...)'. In the future, a tuple will always mean a single key.\n"
     ]
    }
   ],
   "source": [
    "##merging all counts and tpms for transcripts with the same stop codon within a gene\n",
    "datasets = list(combined_read_counts_df.columns[1:-3])\n",
    "print datasets\n",
    "data = []\n",
    "gene_grouped_counts = combined_read_counts_df.groupby('gene_id')\n",
    "gene_stop_grouped_counts = combined_read_counts_df.groupby(('gene_id', 'stop_codon'))\n",
    "for gene_id in combined_read_counts_df['gene_id'].unique():\n",
    "    for stop_codon in gene_grouped_counts.get_group(gene_id)['stop_codon'].unique():\n",
    "        stop_transcript_data = gene_stop_grouped_counts.get_group((gene_id, stop_codon))\n",
    "        tx_ids = '|'.join(sorted(stop_transcript_data['tx_id'].unique()))\n",
    "        data.append([tx_ids] + [stop_transcript_data[dataset].sum() for dataset in datasets])\n",
    "counts_by_stop_df = pd.DataFrame(data, columns = ['tx_ids']+datasets)\n",
    "#counts_by_stop_df['gene_id'] = counts_by_stop_df['tx_ids'].apply(get_gene_id)\n",
    "#counts_by_stop_df['gene_name'] = counts_by_stop_df['tx_ids'].apply(get_gene_name)\n",
    "#counts_by_stop_df['stop_codon'] = counts_by_stop_df['tx_ids'].apply(get_genomic_stop_coords)\n",
    "counts_by_stop_df.to_csv('20200723_summed_stop_codon_counts.tsv', sep='\\t', index=False)\n",
    "\n",
    "data = []\n",
    "gene_grouped_TPM = combined_read_TPM_df.groupby('gene_id')\n",
    "gene_stop_grouped_TPM = combined_read_TPM_df.groupby(('gene_id', 'stop_codon'))\n",
    "for gene_id in combined_read_TPM_df['gene_id'].unique():\n",
    "    for stop_codon in gene_grouped_TPM.get_group(gene_id)['stop_codon'].unique():\n",
    "        stop_transcript_data = gene_stop_grouped_TPM.get_group((gene_id, stop_codon))\n",
    "        tx_ids = '|'.join(sorted(stop_transcript_data['tx_id'].unique()))\n",
    "        data.append([tx_ids] + [stop_transcript_data[dataset].sum() for dataset in datasets])\n",
    "TPM_by_stop_df = pd.DataFrame(data, columns = ['tx_ids']+datasets)\n",
    "TPM_by_stop_df['gene_id'] = TPM_by_stop_df['tx_ids'].apply(get_gene_id)\n",
    "TPM_by_stop_df['gene_name'] = TPM_by_stop_df['tx_ids'].apply(get_gene_name)\n",
    "TPM_by_stop_df['stop_codon'] = TPM_by_stop_df['tx_ids'].apply(get_genomic_stop_coords)\n",
    "TPM_by_stop_df.to_csv('20200723_summed_stop_codon_TPM.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:39.569555Z",
     "start_time": "2020-10-20T02:35:39.565074Z"
    }
   },
   "outputs": [],
   "source": [
    "#Output counts for my favorite genes\n",
    "#TPM_by_stop_df[TPM_by_stop_df['gene_name']=='CCNL1'].to_csv('CCNL1_tpms.tsv', sep ='\\t', index = False )\n",
    "#TPM_by_stop_df[TPM_by_stop_df['gene_name']=='TAF1D'].to_csv('TAF1D_tpms.tsv', sep ='\\t', index = False )\n",
    "#TPM_by_stop_df[TPM_by_stop_df['gene_name']=='GIGYF2'].to_csv('GIGYF2_tpms.tsv', sep ='\\t', index = False )\n",
    "#TPM_by_stop_df[TPM_by_stop_df['gene_name']=='EIF4E2'].to_csv('EIF4E2_tpms.tsv', sep ='\\t', index = False )\n",
    "#TPM_by_stop_df[TPM_by_stop_df['gene_name']=='UGAC_reporter'].to_csv('UGAC_reporter_tpms.tsv', sep ='\\t', index = False )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:39.709785Z",
     "start_time": "2020-10-20T02:35:39.573677Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "UGAC_reporter\n",
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "ENST00000470121.5\n",
      "ENST00000461804.5\n",
      "ENST00000631619.1\n",
      "ENST00000295926.8\n",
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "ENST00000323981.6\n",
      "ENST00000534770.1\n",
      "ENST00000532455.1\n",
      "ENST00000393259.6\n",
      "ENST00000527169.5\n",
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "ENST00000629305.2\n",
      "ENST00000458528.1\n",
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "ENST00000409098.5\n",
      "ENST00000409495.5\n",
      "ENST00000409514.5\n",
      "ENST00000258416.8\n",
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "ENST00000477442.6\n",
      "ENST00000613941.4\n",
      "ENST00000373715.11\n"
     ]
    }
   ],
   "source": [
    "def format_for_prism(subset_df, output_prefix):\n",
    "    treatments = ['DMSO', 'SMG1i']\n",
    "    genotypes = ['WT', 'eIF4E2_4', 'GIGYF2_4']\n",
    "    replicates = ['1', '2']\n",
    "    molecules = ['RNA', 'mono', 'di']\n",
    "    headers = []\n",
    "    for treatment in treatments:\n",
    "        for genotype in genotypes:\n",
    "            for replicate in replicates:\n",
    "                headers.append('%s_%s_%s' % (genotype, treatment, replicate))\n",
    "    for genotype in ['NT1', 'SMG1i', 'eIF4E2', 'GIGYF2']:\n",
    "        for replicate in replicates:\n",
    "            headers.append('%s_%s' % (genotype, replicate))    \n",
    "    print headers\n",
    "    \n",
    "    for tx_id in subset_df['tx_id'].unique():\n",
    "        print tx_id\n",
    "        single_tx =  subset_df[subset_df['tx_id'] == tx_id]\n",
    "        out_file=open('%s_%s.tsv' % (output_prefix, tx_id.replace('|','_')), 'w')\n",
    "        out_file.write('%s_%s\\t%s\\n' % (tx_id, single_tx['stop_codon'].tolist()[0],'\\t'.join(headers)))\n",
    "        for molecule in molecules:\n",
    "            data = []\n",
    "            for header in headers:\n",
    "                if '%s_%s' % (header, molecule) in single_tx.columns:\n",
    "                    data.append(single_tx['%s_%s' % (header, molecule)].tolist()[0])\n",
    "            out_file.write('%s\\t%s\\n' % (molecule, '\\t'.join(['%f' % datum for datum in data])))        \n",
    "        out_file.close()\n",
    "\n",
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='UGAC_reporter'], 'tpms_for_prism/20200823_UGAC_reporter_TPMs')\n",
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='CCNL1'], 'tpms_for_prism/20200823_CCNL1_TPMs')\n",
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='TAF1D'], 'tpms_for_prism/20200823_TAF1D_TPMs')\n",
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='GIGYF2'], 'tpms_for_prism/20200823_GIGYF2_TPMs')\n",
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='EIF4E2'], 'tpms_for_prism/20200823_EIF4E2_TPMs')\n",
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='SRSF3'], 'tpms_for_prism/20200823_SRSF3_TPMs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:39.807526Z",
     "start_time": "2020-10-20T02:35:39.713332Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tx_id</th>\n",
       "      <th>WT_DMSO_1_mono</th>\n",
       "      <th>WT_DMSO_2_mono</th>\n",
       "      <th>WT_SMG1i_1_mono</th>\n",
       "      <th>WT_SMG1i_2_mono</th>\n",
       "      <th>eIF4E2_4_DMSO_1_mono</th>\n",
       "      <th>eIF4E2_4_DMSO_2_mono</th>\n",
       "      <th>eIF4E2_4_SMG1i_1_mono</th>\n",
       "      <th>eIF4E2_4_SMG1i_2_mono</th>\n",
       "      <th>GIGYF2_4_DMSO_1_mono</th>\n",
       "      <th>...</th>\n",
       "      <th>NT1_2_RNA</th>\n",
       "      <th>SMG1i_1_RNA</th>\n",
       "      <th>SMG1i_2_RNA</th>\n",
       "      <th>eIF4E2_1_RNA</th>\n",
       "      <th>eIF4E2_2_RNA</th>\n",
       "      <th>GIGYF2_1_RNA</th>\n",
       "      <th>GIGYF2_2_RNA</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>stop_codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000381655.7</td>\n",
       "      <td>0.079244</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.038273</td>\n",
       "      <td>0.052394</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025955</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.032665</td>\n",
       "      <td>...</td>\n",
       "      <td>0.051967</td>\n",
       "      <td>0.133760</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.067499</td>\n",
       "      <td>0.114234</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000132932.17</td>\n",
       "      <td>ATP8A2</td>\n",
       "      <td>chr13 + 26019983 26019985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000553855.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.269767</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8.554630</td>\n",
       "      <td>3.030680</td>\n",
       "      <td>6.415805</td>\n",
       "      <td>8.497671</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000080815.19</td>\n",
       "      <td>PSEN1</td>\n",
       "      <td>chr14 + 73211856 73211858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000507741.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>4.332386</td>\n",
       "      <td>6.037598</td>\n",
       "      <td>5.519788</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.190198</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000109184.15</td>\n",
       "      <td>DCUN1D4</td>\n",
       "      <td>chr4 + 51848252 51848254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000451080.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000138395.17</td>\n",
       "      <td>CDK15</td>\n",
       "      <td>chr2 + 201807895 201807897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000509622.5</td>\n",
       "      <td>20.144676</td>\n",
       "      <td>9.426756</td>\n",
       "      <td>72.325223</td>\n",
       "      <td>75.045573</td>\n",
       "      <td>10.023676</td>\n",
       "      <td>46.979388</td>\n",
       "      <td>43.451526</td>\n",
       "      <td>64.273158</td>\n",
       "      <td>27.263449</td>\n",
       "      <td>...</td>\n",
       "      <td>19.993439</td>\n",
       "      <td>29.426982</td>\n",
       "      <td>28.428247</td>\n",
       "      <td>25.134720</td>\n",
       "      <td>24.256684</td>\n",
       "      <td>25.671320</td>\n",
       "      <td>30.428565</td>\n",
       "      <td>ENSG00000174720.16</td>\n",
       "      <td>LARP7</td>\n",
       "      <td>chr4 + 112646387 112646389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENST00000567651.5</td>\n",
       "      <td>18.820838</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.434875</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>32.410987</td>\n",
       "      <td>24.502098</td>\n",
       "      <td>60.922129</td>\n",
       "      <td>12.830755</td>\n",
       "      <td>26.693700</td>\n",
       "      <td>...</td>\n",
       "      <td>2.509437</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.822489</td>\n",
       "      <td>5.749014</td>\n",
       "      <td>10.430689</td>\n",
       "      <td>3.910786</td>\n",
       "      <td>4.898560</td>\n",
       "      <td>ENSG00000170100.13</td>\n",
       "      <td>ZNF778</td>\n",
       "      <td>chr16 + 89224743 89224745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000253047.7</td>\n",
       "      <td>13.183857</td>\n",
       "      <td>10.848472</td>\n",
       "      <td>10.862354</td>\n",
       "      <td>8.716825</td>\n",
       "      <td>14.718575</td>\n",
       "      <td>10.823114</td>\n",
       "      <td>11.303603</td>\n",
       "      <td>12.595418</td>\n",
       "      <td>16.144015</td>\n",
       "      <td>...</td>\n",
       "      <td>9.671673</td>\n",
       "      <td>1.555909</td>\n",
       "      <td>10.522071</td>\n",
       "      <td>9.766912</td>\n",
       "      <td>7.851533</td>\n",
       "      <td>6.643918</td>\n",
       "      <td>8.147733</td>\n",
       "      <td>ENSG00000130748.7</td>\n",
       "      <td>TMEM160</td>\n",
       "      <td>chr19 - 47045987 47045989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000373544.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.563753</td>\n",
       "      <td>2.748640</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.658362</td>\n",
       "      <td>5.248733</td>\n",
       "      <td>2.076322</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>3.297825</td>\n",
       "      <td>3.031236</td>\n",
       "      <td>4.070780</td>\n",
       "      <td>3.858440</td>\n",
       "      <td>1.957429</td>\n",
       "      <td>4.118885</td>\n",
       "      <td>6.129696</td>\n",
       "      <td>ENSG00000136933.17</td>\n",
       "      <td>RABEPK</td>\n",
       "      <td>chr9 + 125220566 125220568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENST00000653405.1</td>\n",
       "      <td>10.454812</td>\n",
       "      <td>9.187503</td>\n",
       "      <td>8.655202</td>\n",
       "      <td>6.961902</td>\n",
       "      <td>10.035286</td>\n",
       "      <td>9.926026</td>\n",
       "      <td>10.031100</td>\n",
       "      <td>10.335911</td>\n",
       "      <td>12.893119</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000164897.14</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>chr7 - 151081202 151081204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENST00000520710.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.922637</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.249569</td>\n",
       "      <td>0.368436</td>\n",
       "      <td>1.193265</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.830919</td>\n",
       "      <td>ENSG00000147536.12</td>\n",
       "      <td>GINS4</td>\n",
       "      <td>chr8 + 41537297 41537299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENST00000270538.8</td>\n",
       "      <td>28.925306</td>\n",
       "      <td>26.672905</td>\n",
       "      <td>31.569763</td>\n",
       "      <td>29.199324</td>\n",
       "      <td>20.912322</td>\n",
       "      <td>23.910711</td>\n",
       "      <td>27.411869</td>\n",
       "      <td>28.842510</td>\n",
       "      <td>25.865274</td>\n",
       "      <td>...</td>\n",
       "      <td>74.860906</td>\n",
       "      <td>114.226423</td>\n",
       "      <td>84.987883</td>\n",
       "      <td>97.771734</td>\n",
       "      <td>73.769825</td>\n",
       "      <td>70.564116</td>\n",
       "      <td>66.639058</td>\n",
       "      <td>ENSG00000104980.8</td>\n",
       "      <td>TIMM44</td>\n",
       "      <td>chr19 - 7927187 7927189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000579732.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.985724</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.313236</td>\n",
       "      <td>1.206578</td>\n",
       "      <td>0.305212</td>\n",
       "      <td>0.738049</td>\n",
       "      <td>...</td>\n",
       "      <td>2.649313</td>\n",
       "      <td>4.215564</td>\n",
       "      <td>1.225535</td>\n",
       "      <td>1.877527</td>\n",
       "      <td>2.022479</td>\n",
       "      <td>5.298926</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000133195.11</td>\n",
       "      <td>SLC39A11</td>\n",
       "      <td>chr17 - 72838038 72838040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENST00000471844.6</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.176909</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.279240</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.225934</td>\n",
       "      <td>6.086689</td>\n",
       "      <td>7.833848</td>\n",
       "      <td>5.234769</td>\n",
       "      <td>2.521553</td>\n",
       "      <td>0.812721</td>\n",
       "      <td>8.052507</td>\n",
       "      <td>ENSG00000143437.21</td>\n",
       "      <td>ARNT</td>\n",
       "      <td>chr1 - 150818022 150818024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENST00000395047.6</td>\n",
       "      <td>136.726804</td>\n",
       "      <td>131.959393</td>\n",
       "      <td>122.830889</td>\n",
       "      <td>124.029771</td>\n",
       "      <td>134.362690</td>\n",
       "      <td>136.303981</td>\n",
       "      <td>135.875947</td>\n",
       "      <td>119.539961</td>\n",
       "      <td>152.234074</td>\n",
       "      <td>...</td>\n",
       "      <td>360.508932</td>\n",
       "      <td>310.994821</td>\n",
       "      <td>318.754554</td>\n",
       "      <td>368.019429</td>\n",
       "      <td>380.095472</td>\n",
       "      <td>378.334917</td>\n",
       "      <td>358.650836</td>\n",
       "      <td>ENSG00000213585.11</td>\n",
       "      <td>VDAC1</td>\n",
       "      <td>chr5 - 133972771 133972773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000504963.5</td>\n",
       "      <td>57.448997</td>\n",
       "      <td>80.372457</td>\n",
       "      <td>178.080111</td>\n",
       "      <td>234.230603</td>\n",
       "      <td>227.394447</td>\n",
       "      <td>75.673564</td>\n",
       "      <td>140.505392</td>\n",
       "      <td>193.583192</td>\n",
       "      <td>79.090266</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000143252.15</td>\n",
       "      <td>SDHC</td>\n",
       "      <td>chr1 + 161339584 161339586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENST00000261438.10</td>\n",
       "      <td>8.173133</td>\n",
       "      <td>7.401466</td>\n",
       "      <td>6.812210</td>\n",
       "      <td>9.325486</td>\n",
       "      <td>9.861839</td>\n",
       "      <td>8.514441</td>\n",
       "      <td>10.388636</td>\n",
       "      <td>8.850564</td>\n",
       "      <td>6.616041</td>\n",
       "      <td>...</td>\n",
       "      <td>35.625215</td>\n",
       "      <td>33.189506</td>\n",
       "      <td>34.448755</td>\n",
       "      <td>37.641184</td>\n",
       "      <td>38.138969</td>\n",
       "      <td>39.600825</td>\n",
       "      <td>35.819527</td>\n",
       "      <td>ENSG00000109787.13</td>\n",
       "      <td>KLF3</td>\n",
       "      <td>chr4 + 38697261 38697263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENST00000647410.1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000131951.11</td>\n",
       "      <td>LRRC9</td>\n",
       "      <td>chr14 + 60063360 60063362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENST00000448667.5</td>\n",
       "      <td>1.550243</td>\n",
       "      <td>2.119008</td>\n",
       "      <td>3.112269</td>\n",
       "      <td>3.202133</td>\n",
       "      <td>1.759807</td>\n",
       "      <td>3.234962</td>\n",
       "      <td>5.578859</td>\n",
       "      <td>1.126768</td>\n",
       "      <td>3.058765</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.990681</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000198556.14</td>\n",
       "      <td>ZNF789</td>\n",
       "      <td>chr7 + 99485318 99485320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENST00000357727.7</td>\n",
       "      <td>0.040586</td>\n",
       "      <td>0.041141</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036218</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.068173</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.129849</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000146592.17</td>\n",
       "      <td>CREB5</td>\n",
       "      <td>chr7 + 28819277 28819279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENST00000360104.7</td>\n",
       "      <td>0.314991</td>\n",
       "      <td>0.439459</td>\n",
       "      <td>0.384611</td>\n",
       "      <td>0.333952</td>\n",
       "      <td>0.286441</td>\n",
       "      <td>0.376961</td>\n",
       "      <td>0.327840</td>\n",
       "      <td>0.302748</td>\n",
       "      <td>0.226062</td>\n",
       "      <td>...</td>\n",
       "      <td>5.996722</td>\n",
       "      <td>9.373054</td>\n",
       "      <td>8.747670</td>\n",
       "      <td>6.122039</td>\n",
       "      <td>6.403483</td>\n",
       "      <td>7.536654</td>\n",
       "      <td>9.051557</td>\n",
       "      <td>ENSG00000055609.18</td>\n",
       "      <td>KMT2C</td>\n",
       "      <td>chr7 - 152138784 152138786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENST00000578681.5</td>\n",
       "      <td>9.315369</td>\n",
       "      <td>9.568756</td>\n",
       "      <td>10.249660</td>\n",
       "      <td>11.382963</td>\n",
       "      <td>9.252567</td>\n",
       "      <td>11.365201</td>\n",
       "      <td>10.617787</td>\n",
       "      <td>10.051683</td>\n",
       "      <td>12.777803</td>\n",
       "      <td>...</td>\n",
       "      <td>42.954602</td>\n",
       "      <td>79.440383</td>\n",
       "      <td>85.802850</td>\n",
       "      <td>41.805900</td>\n",
       "      <td>43.227448</td>\n",
       "      <td>45.617779</td>\n",
       "      <td>44.145298</td>\n",
       "      <td>ENSG00000198231.13</td>\n",
       "      <td>DDX42</td>\n",
       "      <td>chr17 + 63818396 63818398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENST00000382452.6</td>\n",
       "      <td>48.440174</td>\n",
       "      <td>48.027534</td>\n",
       "      <td>47.127674</td>\n",
       "      <td>48.632560</td>\n",
       "      <td>40.007920</td>\n",
       "      <td>42.953439</td>\n",
       "      <td>43.029356</td>\n",
       "      <td>41.955448</td>\n",
       "      <td>44.593147</td>\n",
       "      <td>...</td>\n",
       "      <td>125.115983</td>\n",
       "      <td>134.221344</td>\n",
       "      <td>126.428571</td>\n",
       "      <td>124.271948</td>\n",
       "      <td>129.525861</td>\n",
       "      <td>150.356557</td>\n",
       "      <td>146.278845</td>\n",
       "      <td>ENSG00000071127.17</td>\n",
       "      <td>WDR1</td>\n",
       "      <td>chr4 - 10075378 10075380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENST00000357988.9</td>\n",
       "      <td>1.722556</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.373756</td>\n",
       "      <td>2.508354</td>\n",
       "      <td>1.264120</td>\n",
       "      <td>1.571929</td>\n",
       "      <td>0.990750</td>\n",
       "      <td>1.881946</td>\n",
       "      <td>1.441474</td>\n",
       "      <td>...</td>\n",
       "      <td>3.629582</td>\n",
       "      <td>4.275742</td>\n",
       "      <td>4.155592</td>\n",
       "      <td>5.269406</td>\n",
       "      <td>4.598725</td>\n",
       "      <td>7.323335</td>\n",
       "      <td>6.509015</td>\n",
       "      <td>ENSG00000172943.20</td>\n",
       "      <td>PHF8</td>\n",
       "      <td>chrX - 53939158 53939160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENST00000657480.1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.709860</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>8.458539</td>\n",
       "      <td>8.476313</td>\n",
       "      <td>12.606013</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>9.323937</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.196193</td>\n",
       "      <td>ENSG00000163681.16</td>\n",
       "      <td>SLMAP</td>\n",
       "      <td>chr3 + 57883981 57883983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENST00000521156.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.420477</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000135297.16</td>\n",
       "      <td>MTO1</td>\n",
       "      <td>chr6 + 73500585 73500587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENST00000309352.4</td>\n",
       "      <td>27.766527</td>\n",
       "      <td>26.211011</td>\n",
       "      <td>28.778842</td>\n",
       "      <td>25.242949</td>\n",
       "      <td>26.293289</td>\n",
       "      <td>28.088964</td>\n",
       "      <td>27.823143</td>\n",
       "      <td>26.275183</td>\n",
       "      <td>32.094658</td>\n",
       "      <td>...</td>\n",
       "      <td>73.797790</td>\n",
       "      <td>90.026288</td>\n",
       "      <td>74.180485</td>\n",
       "      <td>61.200612</td>\n",
       "      <td>76.339861</td>\n",
       "      <td>57.798491</td>\n",
       "      <td>58.968111</td>\n",
       "      <td>ENSG00000204316.13</td>\n",
       "      <td>MRPL38</td>\n",
       "      <td>chr17 - 75898850 75898852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENST00000651723.1</td>\n",
       "      <td>101.472666</td>\n",
       "      <td>58.490537</td>\n",
       "      <td>150.504311</td>\n",
       "      <td>153.559623</td>\n",
       "      <td>101.939670</td>\n",
       "      <td>89.626866</td>\n",
       "      <td>220.587325</td>\n",
       "      <td>113.554652</td>\n",
       "      <td>65.632025</td>\n",
       "      <td>...</td>\n",
       "      <td>0.845742</td>\n",
       "      <td>1.753922</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.538182</td>\n",
       "      <td>ENSG00000113522.14</td>\n",
       "      <td>RAD50</td>\n",
       "      <td>chr5 + 132557466 132557468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENST00000260600.9</td>\n",
       "      <td>1.175992</td>\n",
       "      <td>0.936597</td>\n",
       "      <td>0.956274</td>\n",
       "      <td>1.199989</td>\n",
       "      <td>0.941694</td>\n",
       "      <td>0.990575</td>\n",
       "      <td>0.972757</td>\n",
       "      <td>1.015934</td>\n",
       "      <td>1.054210</td>\n",
       "      <td>...</td>\n",
       "      <td>12.342717</td>\n",
       "      <td>11.754146</td>\n",
       "      <td>13.494555</td>\n",
       "      <td>10.342743</td>\n",
       "      <td>9.274623</td>\n",
       "      <td>10.202567</td>\n",
       "      <td>9.349523</td>\n",
       "      <td>ENSG00000138031.14</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>chr2 - 24819932 24819934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENST00000371442.4</td>\n",
       "      <td>12.085520</td>\n",
       "      <td>10.793870</td>\n",
       "      <td>9.150051</td>\n",
       "      <td>8.098615</td>\n",
       "      <td>10.511692</td>\n",
       "      <td>10.699298</td>\n",
       "      <td>8.826345</td>\n",
       "      <td>8.103191</td>\n",
       "      <td>10.906245</td>\n",
       "      <td>...</td>\n",
       "      <td>35.909321</td>\n",
       "      <td>34.973252</td>\n",
       "      <td>22.806510</td>\n",
       "      <td>37.523031</td>\n",
       "      <td>32.775676</td>\n",
       "      <td>46.935431</td>\n",
       "      <td>31.395822</td>\n",
       "      <td>ENSG00000125352.6</td>\n",
       "      <td>RNF113A</td>\n",
       "      <td>chrX - 119870582 119870584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENST00000572944.1</td>\n",
       "      <td>0.669492</td>\n",
       "      <td>0.685768</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.188844</td>\n",
       "      <td>0.494482</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.143450</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.213753</td>\n",
       "      <td>...</td>\n",
       "      <td>3.795869</td>\n",
       "      <td>4.276929</td>\n",
       "      <td>6.673128</td>\n",
       "      <td>2.569567</td>\n",
       "      <td>2.906855</td>\n",
       "      <td>3.490577</td>\n",
       "      <td>2.376713</td>\n",
       "      <td>ENSG00000153933.10</td>\n",
       "      <td>DGKE</td>\n",
       "      <td>chr17 + 56862264 56862266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24521</th>\n",
       "      <td>ENST00000625106.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.868030</td>\n",
       "      <td>5.135394</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.688782</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.869780</td>\n",
       "      <td>5.694820</td>\n",
       "      <td>ENSG00000269405.7</td>\n",
       "      <td>NXF2</td>\n",
       "      <td>chrX + 102326512 102326514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24522</th>\n",
       "      <td>ENST00000661438.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>37.011236</td>\n",
       "      <td>24.901065</td>\n",
       "      <td>54.641657</td>\n",
       "      <td>51.173751</td>\n",
       "      <td>32.729327</td>\n",
       "      <td>28.876636</td>\n",
       "      <td>50.886729</td>\n",
       "      <td>ENSG00000285920.2</td>\n",
       "      <td>AC087721.2</td>\n",
       "      <td>chr15 + 41380184 41380186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24523</th>\n",
       "      <td>ENST00000639028.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3836.556580</td>\n",
       "      <td>3209.366915</td>\n",
       "      <td>3088.002731</td>\n",
       "      <td>3736.286060</td>\n",
       "      <td>3639.802149</td>\n",
       "      <td>3575.211507</td>\n",
       "      <td>3639.955752</td>\n",
       "      <td>ENSG00000236362.8</td>\n",
       "      <td>GAGE12F</td>\n",
       "      <td>chrX + 49568112 49568114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24524</th>\n",
       "      <td>ENST00000637376.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000261832.6</td>\n",
       "      <td>AC138894.1</td>\n",
       "      <td>chr16 - 28479730 28479732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24525</th>\n",
       "      <td>ENST00000622306.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>34.586982</td>\n",
       "      <td>35.818655</td>\n",
       "      <td>42.082036</td>\n",
       "      <td>32.945114</td>\n",
       "      <td>43.001253</td>\n",
       "      <td>28.872605</td>\n",
       "      <td>22.747973</td>\n",
       "      <td>ENSG00000185164.14</td>\n",
       "      <td>NOMO2</td>\n",
       "      <td>chr16 - 18500436 18500438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24526</th>\n",
       "      <td>ENST00000511629.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.067019</td>\n",
       "      <td>0.999311</td>\n",
       "      <td>0.573203</td>\n",
       "      <td>0.772236</td>\n",
       "      <td>0.433510</td>\n",
       "      <td>1.478210</td>\n",
       "      <td>1.367791</td>\n",
       "      <td>ENSG00000183474.15</td>\n",
       "      <td>GTF2H2C</td>\n",
       "      <td>chr5 + 69566986 69566988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24527</th>\n",
       "      <td>ENST00000342136.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.021548</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000159110.20</td>\n",
       "      <td>IFNAR2</td>\n",
       "      <td>chr21 + 33263498 33263500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24528</th>\n",
       "      <td>ENST00000557565.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>15.410169</td>\n",
       "      <td>12.854871</td>\n",
       "      <td>18.418913</td>\n",
       "      <td>16.502977</td>\n",
       "      <td>12.110023</td>\n",
       "      <td>14.354768</td>\n",
       "      <td>17.718008</td>\n",
       "      <td>ENSG00000258790.1</td>\n",
       "      <td>AL121594.1</td>\n",
       "      <td>chr14 + 35273564 35273566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24529</th>\n",
       "      <td>ENST00000569360.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.594774</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.077352</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000261459.1</td>\n",
       "      <td>AC002310.4</td>\n",
       "      <td>chr16 - 30532499 30532501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24530</th>\n",
       "      <td>ENST00000644653.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>10.143192</td>\n",
       "      <td>8.768513</td>\n",
       "      <td>6.084425</td>\n",
       "      <td>8.207644</td>\n",
       "      <td>9.767945</td>\n",
       "      <td>5.746016</td>\n",
       "      <td>7.938302</td>\n",
       "      <td>ENSG00000165060.14</td>\n",
       "      <td>FXN</td>\n",
       "      <td>chr9 + 69064948 69064950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24531</th>\n",
       "      <td>ENST00000355057.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7143.761224</td>\n",
       "      <td>6202.428007</td>\n",
       "      <td>6777.244725</td>\n",
       "      <td>7032.415311</td>\n",
       "      <td>6123.059355</td>\n",
       "      <td>5507.448800</td>\n",
       "      <td>5569.149479</td>\n",
       "      <td>ENSG00000197238.5</td>\n",
       "      <td>H4C11</td>\n",
       "      <td>chr6 + 27824434 27824436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24532</th>\n",
       "      <td>ENST00000322209.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>130.447004</td>\n",
       "      <td>120.712251</td>\n",
       "      <td>100.704702</td>\n",
       "      <td>138.819206</td>\n",
       "      <td>123.476630</td>\n",
       "      <td>94.945869</td>\n",
       "      <td>101.415155</td>\n",
       "      <td>ENSG00000177144.7</td>\n",
       "      <td>NUDT4B</td>\n",
       "      <td>chr1 + 148749496 148749498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24533</th>\n",
       "      <td>ENST00000329516.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.141864</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.971552</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000183054.11</td>\n",
       "      <td>RGPD6</td>\n",
       "      <td>chr2 - 110515611 110515613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24534</th>\n",
       "      <td>ENST00000593751.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3.586274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>9.290016</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.425039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000179304.16</td>\n",
       "      <td>FAM156B</td>\n",
       "      <td>chrX + 52907865 52907867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24535</th>\n",
       "      <td>ENST00000223321.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>32.587163</td>\n",
       "      <td>14.936309</td>\n",
       "      <td>12.583842</td>\n",
       "      <td>35.466212</td>\n",
       "      <td>41.974285</td>\n",
       "      <td>4.812858</td>\n",
       "      <td>29.691841</td>\n",
       "      <td>ENSG00000106588.11</td>\n",
       "      <td>PSMA2</td>\n",
       "      <td>chr7 - 42917574 42917576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24536</th>\n",
       "      <td>ENST00000595065.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>82.705383</td>\n",
       "      <td>78.110219</td>\n",
       "      <td>83.750816</td>\n",
       "      <td>79.283693</td>\n",
       "      <td>109.467617</td>\n",
       "      <td>104.728122</td>\n",
       "      <td>107.239231</td>\n",
       "      <td>ENSG00000267978.5</td>\n",
       "      <td>MAGEA9B</td>\n",
       "      <td>chrX - 149582232 149582234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24537</th>\n",
       "      <td>ENST00000578794.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8.373478</td>\n",
       "      <td>18.674512</td>\n",
       "      <td>24.160319</td>\n",
       "      <td>5.764022</td>\n",
       "      <td>0.861763</td>\n",
       "      <td>3.102230</td>\n",
       "      <td>2.426199</td>\n",
       "      <td>ENSG00000266086.3</td>\n",
       "      <td>AC015813.2</td>\n",
       "      <td>chr17 - 58005747 58005749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24538</th>\n",
       "      <td>ENST00000674062.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000288558.1</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>chr7 + 107575220 107575222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24539</th>\n",
       "      <td>ENST00000631420.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.887812</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000256977.12</td>\n",
       "      <td>LIMS3</td>\n",
       "      <td>chr2 + 109919553 109919555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24540</th>\n",
       "      <td>ENST00000248923.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000286070.1</td>\n",
       "      <td>GGT1</td>\n",
       "      <td>chr22 + 24628837 24628839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24541</th>\n",
       "      <td>ENST00000317633.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3.313570</td>\n",
       "      <td>16.368510</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.512578</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000172058.15</td>\n",
       "      <td>SERF1A</td>\n",
       "      <td>chr5 + 70907804 70907806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24542</th>\n",
       "      <td>ENST00000564219.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>9.741034</td>\n",
       "      <td>30.930332</td>\n",
       "      <td>27.082895</td>\n",
       "      <td>12.679435</td>\n",
       "      <td>22.650132</td>\n",
       "      <td>21.824424</td>\n",
       "      <td>3.148545</td>\n",
       "      <td>ENSG00000198064.13</td>\n",
       "      <td>NPIPB13</td>\n",
       "      <td>chr16 - 30240213 30240215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24543</th>\n",
       "      <td>ENST00000300035.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>53.602335</td>\n",
       "      <td>46.127352</td>\n",
       "      <td>37.791175</td>\n",
       "      <td>39.113089</td>\n",
       "      <td>38.935980</td>\n",
       "      <td>36.675994</td>\n",
       "      <td>37.184680</td>\n",
       "      <td>ENSG00000166803.13</td>\n",
       "      <td>PCLAF</td>\n",
       "      <td>chr15 - 64366030 64366032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24544</th>\n",
       "      <td>ENST00000324103.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>6.927750</td>\n",
       "      <td>13.984064</td>\n",
       "      <td>13.506764</td>\n",
       "      <td>9.815328</td>\n",
       "      <td>8.662501</td>\n",
       "      <td>9.859112</td>\n",
       "      <td>7.926904</td>\n",
       "      <td>ENSG00000092098.17</td>\n",
       "      <td>RNF31</td>\n",
       "      <td>chr14 + 24160571 24160573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24545</th>\n",
       "      <td>ENST00000360423.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000213648.11</td>\n",
       "      <td>SULT1A4</td>\n",
       "      <td>chr16 + 29464573 29464575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24546</th>\n",
       "      <td>ENST00000607355.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4282.289793</td>\n",
       "      <td>3230.788916</td>\n",
       "      <td>3332.317148</td>\n",
       "      <td>4012.867833</td>\n",
       "      <td>3893.817815</td>\n",
       "      <td>3561.852333</td>\n",
       "      <td>3615.970743</td>\n",
       "      <td>ENSG00000272196.2</td>\n",
       "      <td>H2AC19</td>\n",
       "      <td>chr1 + 149851504 149851506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24547</th>\n",
       "      <td>ENST00000304298.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3.302721</td>\n",
       "      <td>1.062636</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.952927</td>\n",
       "      <td>2.144938</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.112928</td>\n",
       "      <td>ENSG00000170276.6</td>\n",
       "      <td>HSPB2</td>\n",
       "      <td>chr11 + 111913893 111913895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24548</th>\n",
       "      <td>ENST00000369562.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000272514.6</td>\n",
       "      <td>CFAP206</td>\n",
       "      <td>chr6 + 87464248 87464250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24549</th>\n",
       "      <td>ENST00000596542.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000269307.1</td>\n",
       "      <td>AC010463.1</td>\n",
       "      <td>chr19 + 17273943 17273945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24550</th>\n",
       "      <td>ENST00000572919.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.437877</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000262304.2</td>\n",
       "      <td>AC027796.3</td>\n",
       "      <td>chr17 - 3610560 3610562</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24551 rows × 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    tx_id  WT_DMSO_1_mono  WT_DMSO_2_mono  WT_SMG1i_1_mono  \\\n",
       "0       ENST00000381655.7        0.079244        0.000000         0.038273   \n",
       "1       ENST00000553855.5        0.000000        0.000000         0.000000   \n",
       "2       ENST00000507741.5        0.000000        0.000000         0.000000   \n",
       "3       ENST00000451080.5        0.000000        0.000000         0.000000   \n",
       "4       ENST00000509622.5       20.144676        9.426756        72.325223   \n",
       "5       ENST00000567651.5       18.820838        0.000000         7.434875   \n",
       "6       ENST00000253047.7       13.183857       10.848472        10.862354   \n",
       "7       ENST00000373544.5        0.000000        0.000000         2.563753   \n",
       "8       ENST00000653405.1       10.454812        9.187503         8.655202   \n",
       "9       ENST00000520710.5        0.000000        0.000000         0.000000   \n",
       "10      ENST00000270538.8       28.925306       26.672905        31.569763   \n",
       "11      ENST00000579732.5        0.000000        0.000000         1.985724   \n",
       "12      ENST00000471844.6        0.000000        0.000000         1.176909   \n",
       "13      ENST00000395047.6      136.726804      131.959393       122.830889   \n",
       "14      ENST00000504963.5       57.448997       80.372457       178.080111   \n",
       "15     ENST00000261438.10        8.173133        7.401466         6.812210   \n",
       "16      ENST00000647410.1        0.000000        0.000000         0.000000   \n",
       "17      ENST00000448667.5        1.550243        2.119008         3.112269   \n",
       "18      ENST00000357727.7        0.040586        0.041141         0.000000   \n",
       "19      ENST00000360104.7        0.314991        0.439459         0.384611   \n",
       "20      ENST00000578681.5        9.315369        9.568756        10.249660   \n",
       "21      ENST00000382452.6       48.440174       48.027534        47.127674   \n",
       "22      ENST00000357988.9        1.722556        0.000000         2.373756   \n",
       "23      ENST00000657480.1        0.000000        0.000000         0.000000   \n",
       "24      ENST00000521156.5        0.000000        0.000000         0.000000   \n",
       "25      ENST00000309352.4       27.766527       26.211011        28.778842   \n",
       "26      ENST00000651723.1      101.472666       58.490537       150.504311   \n",
       "27      ENST00000260600.9        1.175992        0.936597         0.956274   \n",
       "28      ENST00000371442.4       12.085520       10.793870         9.150051   \n",
       "29      ENST00000572944.1        0.669492        0.685768         0.000000   \n",
       "...                   ...             ...             ...              ...   \n",
       "24521   ENST00000625106.4             NaN             NaN              NaN   \n",
       "24522   ENST00000661438.1             NaN             NaN              NaN   \n",
       "24523   ENST00000639028.1             NaN             NaN              NaN   \n",
       "24524   ENST00000637376.1             NaN             NaN              NaN   \n",
       "24525   ENST00000622306.4             NaN             NaN              NaN   \n",
       "24526   ENST00000511629.6             NaN             NaN              NaN   \n",
       "24527   ENST00000342136.9             NaN             NaN              NaN   \n",
       "24528   ENST00000557565.1             NaN             NaN              NaN   \n",
       "24529   ENST00000569360.1             NaN             NaN              NaN   \n",
       "24530   ENST00000644653.1             NaN             NaN              NaN   \n",
       "24531   ENST00000355057.3             NaN             NaN              NaN   \n",
       "24532   ENST00000322209.4             NaN             NaN              NaN   \n",
       "24533   ENST00000329516.7             NaN             NaN              NaN   \n",
       "24534   ENST00000593751.7             NaN             NaN              NaN   \n",
       "24535   ENST00000223321.9             NaN             NaN              NaN   \n",
       "24536   ENST00000595065.5             NaN             NaN              NaN   \n",
       "24537   ENST00000578794.2             NaN             NaN              NaN   \n",
       "24538   ENST00000674062.1             NaN             NaN              NaN   \n",
       "24539   ENST00000631420.1             NaN             NaN              NaN   \n",
       "24540   ENST00000248923.9             NaN             NaN              NaN   \n",
       "24541  ENST00000317633.13             NaN             NaN              NaN   \n",
       "24542   ENST00000564219.1             NaN             NaN              NaN   \n",
       "24543   ENST00000300035.9             NaN             NaN              NaN   \n",
       "24544  ENST00000324103.11             NaN             NaN              NaN   \n",
       "24545  ENST00000360423.12             NaN             NaN              NaN   \n",
       "24546   ENST00000607355.2             NaN             NaN              NaN   \n",
       "24547   ENST00000304298.4             NaN             NaN              NaN   \n",
       "24548   ENST00000369562.9             NaN             NaN              NaN   \n",
       "24549   ENST00000596542.1             NaN             NaN              NaN   \n",
       "24550   ENST00000572919.1             NaN             NaN              NaN   \n",
       "\n",
       "       WT_SMG1i_2_mono  eIF4E2_4_DMSO_1_mono  eIF4E2_4_DMSO_2_mono  \\\n",
       "0             0.052394              0.000000              0.000000   \n",
       "1             0.000000              0.000000              0.000000   \n",
       "2             0.000000              0.000000              0.000000   \n",
       "3             0.000000              0.000000              0.000000   \n",
       "4            75.045573             10.023676             46.979388   \n",
       "5             0.000000             32.410987             24.502098   \n",
       "6             8.716825             14.718575             10.823114   \n",
       "7             2.748640              0.000000              2.658362   \n",
       "8             6.961902             10.035286              9.926026   \n",
       "9             0.000000              0.000000              0.000000   \n",
       "10           29.199324             20.912322             23.910711   \n",
       "11            0.000000              0.000000              0.313236   \n",
       "12            0.000000              0.000000              0.000000   \n",
       "13          124.029771            134.362690            136.303981   \n",
       "14          234.230603            227.394447             75.673564   \n",
       "15            9.325486              9.861839              8.514441   \n",
       "16            0.000000              0.000000              0.000000   \n",
       "17            3.202133              1.759807              3.234962   \n",
       "18            0.000000              0.000000              0.000000   \n",
       "19            0.333952              0.286441              0.376961   \n",
       "20           11.382963              9.252567             11.365201   \n",
       "21           48.632560             40.007920             42.953439   \n",
       "22            2.508354              1.264120              1.571929   \n",
       "23            0.000000              0.000000              0.000000   \n",
       "24            0.000000              0.420477              0.000000   \n",
       "25           25.242949             26.293289             28.088964   \n",
       "26          153.559623            101.939670             89.626866   \n",
       "27            1.199989              0.941694              0.990575   \n",
       "28            8.098615             10.511692             10.699298   \n",
       "29            0.188844              0.494482              0.000000   \n",
       "...                ...                   ...                   ...   \n",
       "24521              NaN                   NaN                   NaN   \n",
       "24522              NaN                   NaN                   NaN   \n",
       "24523              NaN                   NaN                   NaN   \n",
       "24524              NaN                   NaN                   NaN   \n",
       "24525              NaN                   NaN                   NaN   \n",
       "24526              NaN                   NaN                   NaN   \n",
       "24527              NaN                   NaN                   NaN   \n",
       "24528              NaN                   NaN                   NaN   \n",
       "24529              NaN                   NaN                   NaN   \n",
       "24530              NaN                   NaN                   NaN   \n",
       "24531              NaN                   NaN                   NaN   \n",
       "24532              NaN                   NaN                   NaN   \n",
       "24533              NaN                   NaN                   NaN   \n",
       "24534              NaN                   NaN                   NaN   \n",
       "24535              NaN                   NaN                   NaN   \n",
       "24536              NaN                   NaN                   NaN   \n",
       "24537              NaN                   NaN                   NaN   \n",
       "24538              NaN                   NaN                   NaN   \n",
       "24539              NaN                   NaN                   NaN   \n",
       "24540              NaN                   NaN                   NaN   \n",
       "24541              NaN                   NaN                   NaN   \n",
       "24542              NaN                   NaN                   NaN   \n",
       "24543              NaN                   NaN                   NaN   \n",
       "24544              NaN                   NaN                   NaN   \n",
       "24545              NaN                   NaN                   NaN   \n",
       "24546              NaN                   NaN                   NaN   \n",
       "24547              NaN                   NaN                   NaN   \n",
       "24548              NaN                   NaN                   NaN   \n",
       "24549              NaN                   NaN                   NaN   \n",
       "24550              NaN                   NaN                   NaN   \n",
       "\n",
       "       eIF4E2_4_SMG1i_1_mono  eIF4E2_4_SMG1i_2_mono  GIGYF2_4_DMSO_1_mono  \\\n",
       "0                   0.025955               0.000000              0.032665   \n",
       "1                   0.000000               0.000000              0.269767   \n",
       "2                   0.000000               0.000000              0.000000   \n",
       "3                   0.000000               0.000000              0.000000   \n",
       "4                  43.451526              64.273158             27.263449   \n",
       "5                  60.922129              12.830755             26.693700   \n",
       "6                  11.303603              12.595418             16.144015   \n",
       "7                   5.248733               2.076322              0.000000   \n",
       "8                  10.031100              10.335911             12.893119   \n",
       "9                   0.000000               0.000000              0.000000   \n",
       "10                 27.411869              28.842510             25.865274   \n",
       "11                  1.206578               0.305212              0.738049   \n",
       "12                  0.279240               0.000000              0.000000   \n",
       "13                135.875947             119.539961            152.234074   \n",
       "14                140.505392             193.583192             79.090266   \n",
       "15                 10.388636               8.850564              6.616041   \n",
       "16                  0.000000               0.000000              0.000000   \n",
       "17                  5.578859               1.126768              3.058765   \n",
       "18                  0.000000               0.036218              0.000000   \n",
       "19                  0.327840               0.302748              0.226062   \n",
       "20                 10.617787              10.051683             12.777803   \n",
       "21                 43.029356              41.955448             44.593147   \n",
       "22                  0.990750               1.881946              1.441474   \n",
       "23                  0.000000               0.709860              0.000000   \n",
       "24                  0.000000               0.000000              0.000000   \n",
       "25                 27.823143              26.275183             32.094658   \n",
       "26                220.587325             113.554652             65.632025   \n",
       "27                  0.972757               1.015934              1.054210   \n",
       "28                  8.826345               8.103191             10.906245   \n",
       "29                  1.143450               0.000000              0.213753   \n",
       "...                      ...                    ...                   ...   \n",
       "24521                    NaN                    NaN                   NaN   \n",
       "24522                    NaN                    NaN                   NaN   \n",
       "24523                    NaN                    NaN                   NaN   \n",
       "24524                    NaN                    NaN                   NaN   \n",
       "24525                    NaN                    NaN                   NaN   \n",
       "24526                    NaN                    NaN                   NaN   \n",
       "24527                    NaN                    NaN                   NaN   \n",
       "24528                    NaN                    NaN                   NaN   \n",
       "24529                    NaN                    NaN                   NaN   \n",
       "24530                    NaN                    NaN                   NaN   \n",
       "24531                    NaN                    NaN                   NaN   \n",
       "24532                    NaN                    NaN                   NaN   \n",
       "24533                    NaN                    NaN                   NaN   \n",
       "24534                    NaN                    NaN                   NaN   \n",
       "24535                    NaN                    NaN                   NaN   \n",
       "24536                    NaN                    NaN                   NaN   \n",
       "24537                    NaN                    NaN                   NaN   \n",
       "24538                    NaN                    NaN                   NaN   \n",
       "24539                    NaN                    NaN                   NaN   \n",
       "24540                    NaN                    NaN                   NaN   \n",
       "24541                    NaN                    NaN                   NaN   \n",
       "24542                    NaN                    NaN                   NaN   \n",
       "24543                    NaN                    NaN                   NaN   \n",
       "24544                    NaN                    NaN                   NaN   \n",
       "24545                    NaN                    NaN                   NaN   \n",
       "24546                    NaN                    NaN                   NaN   \n",
       "24547                    NaN                    NaN                   NaN   \n",
       "24548                    NaN                    NaN                   NaN   \n",
       "24549                    NaN                    NaN                   NaN   \n",
       "24550                    NaN                    NaN                   NaN   \n",
       "\n",
       "                  ...                 NT1_2_RNA  SMG1i_1_RNA  SMG1i_2_RNA  \\\n",
       "0                 ...                  0.051967     0.133760     0.000000   \n",
       "1                 ...                  0.000000     0.000000     8.554630   \n",
       "2                 ...                  4.332386     6.037598     5.519788   \n",
       "3                 ...                  0.000000     0.000000     0.000000   \n",
       "4                 ...                 19.993439    29.426982    28.428247   \n",
       "5                 ...                  2.509437     0.000000     1.822489   \n",
       "6                 ...                  9.671673     1.555909    10.522071   \n",
       "7                 ...                  3.297825     3.031236     4.070780   \n",
       "8                 ...                  0.000000     0.000000     0.000000   \n",
       "9                 ...                  0.922637     0.000000     2.249569   \n",
       "10                ...                 74.860906   114.226423    84.987883   \n",
       "11                ...                  2.649313     4.215564     1.225535   \n",
       "12                ...                  5.225934     6.086689     7.833848   \n",
       "13                ...                360.508932   310.994821   318.754554   \n",
       "14                ...                  0.000000     0.000000     0.000000   \n",
       "15                ...                 35.625215    33.189506    34.448755   \n",
       "16                ...                  0.000000     0.000000     0.000000   \n",
       "17                ...                  0.000000     0.000000     2.990681   \n",
       "18                ...                  0.000000     0.000000     0.000000   \n",
       "19                ...                  5.996722     9.373054     8.747670   \n",
       "20                ...                 42.954602    79.440383    85.802850   \n",
       "21                ...                125.115983   134.221344   126.428571   \n",
       "22                ...                  3.629582     4.275742     4.155592   \n",
       "23                ...                  8.458539     8.476313    12.606013   \n",
       "24                ...                  0.000000     0.000000     0.000000   \n",
       "25                ...                 73.797790    90.026288    74.180485   \n",
       "26                ...                  0.845742     1.753922     0.000000   \n",
       "27                ...                 12.342717    11.754146    13.494555   \n",
       "28                ...                 35.909321    34.973252    22.806510   \n",
       "29                ...                  3.795869     4.276929     6.673128   \n",
       "...               ...                       ...          ...          ...   \n",
       "24521             ...                  5.868030     5.135394     0.000000   \n",
       "24522             ...                 37.011236    24.901065    54.641657   \n",
       "24523             ...               3836.556580  3209.366915  3088.002731   \n",
       "24524             ...                  0.000000     0.000000     0.000000   \n",
       "24525             ...                 34.586982    35.818655    42.082036   \n",
       "24526             ...                  1.067019     0.999311     0.573203   \n",
       "24527             ...                  0.000000     0.000000     2.021548   \n",
       "24528             ...                 15.410169    12.854871    18.418913   \n",
       "24529             ...                  0.000000     0.594774     0.000000   \n",
       "24530             ...                 10.143192     8.768513     6.084425   \n",
       "24531             ...               7143.761224  6202.428007  6777.244725   \n",
       "24532             ...                130.447004   120.712251   100.704702   \n",
       "24533             ...                  0.000000     0.000000     0.141864   \n",
       "24534             ...                  3.586274     0.000000     0.000000   \n",
       "24535             ...                 32.587163    14.936309    12.583842   \n",
       "24536             ...                 82.705383    78.110219    83.750816   \n",
       "24537             ...                  8.373478    18.674512    24.160319   \n",
       "24538             ...                  0.000000     0.000000     0.000000   \n",
       "24539             ...                  0.000000     0.000000     1.887812   \n",
       "24540             ...                  0.000000     0.000000     0.000000   \n",
       "24541             ...                  3.313570    16.368510     0.000000   \n",
       "24542             ...                  9.741034    30.930332    27.082895   \n",
       "24543             ...                 53.602335    46.127352    37.791175   \n",
       "24544             ...                  6.927750    13.984064    13.506764   \n",
       "24545             ...                  0.000000     0.000000     0.000000   \n",
       "24546             ...               4282.289793  3230.788916  3332.317148   \n",
       "24547             ...                  3.302721     1.062636     0.000000   \n",
       "24548             ...                  0.000000     0.000000     0.000000   \n",
       "24549             ...                  0.000000     0.000000     0.000000   \n",
       "24550             ...                  0.000000     0.000000     0.437877   \n",
       "\n",
       "       eIF4E2_1_RNA  eIF4E2_2_RNA  GIGYF2_1_RNA  GIGYF2_2_RNA  \\\n",
       "0          0.000000      0.067499      0.114234      0.000000   \n",
       "1          3.030680      6.415805      8.497671      0.000000   \n",
       "2          0.000000      0.000000      2.190198      0.000000   \n",
       "3          0.000000      0.000000      0.000000      0.000000   \n",
       "4         25.134720     24.256684     25.671320     30.428565   \n",
       "5          5.749014     10.430689      3.910786      4.898560   \n",
       "6          9.766912      7.851533      6.643918      8.147733   \n",
       "7          3.858440      1.957429      4.118885      6.129696   \n",
       "8          0.000000      0.000000      0.000000      0.000000   \n",
       "9          0.368436      1.193265      0.000000      0.830919   \n",
       "10        97.771734     73.769825     70.564116     66.639058   \n",
       "11         1.877527      2.022479      5.298926      0.000000   \n",
       "12         5.234769      2.521553      0.812721      8.052507   \n",
       "13       368.019429    380.095472    378.334917    358.650836   \n",
       "14         0.000000      0.000000      0.000000      0.000000   \n",
       "15        37.641184     38.138969     39.600825     35.819527   \n",
       "16         0.000000      0.000000      0.000000      0.000000   \n",
       "17         0.000000      0.000000      0.000000      0.000000   \n",
       "18         0.068173      0.000000      0.129849      0.000000   \n",
       "19         6.122039      6.403483      7.536654      9.051557   \n",
       "20        41.805900     43.227448     45.617779     44.145298   \n",
       "21       124.271948    129.525861    150.356557    146.278845   \n",
       "22         5.269406      4.598725      7.323335      6.509015   \n",
       "23         0.000000      9.323937      0.000000      3.196193   \n",
       "24         0.000000      0.000000      0.000000      0.000000   \n",
       "25        61.200612     76.339861     57.798491     58.968111   \n",
       "26         0.000000      0.000000      0.000000      2.538182   \n",
       "27        10.342743      9.274623     10.202567      9.349523   \n",
       "28        37.523031     32.775676     46.935431     31.395822   \n",
       "29         2.569567      2.906855      3.490577      2.376713   \n",
       "...             ...           ...           ...           ...   \n",
       "24521      5.688782      0.000000      3.869780      5.694820   \n",
       "24522     51.173751     32.729327     28.876636     50.886729   \n",
       "24523   3736.286060   3639.802149   3575.211507   3639.955752   \n",
       "24524      0.000000      0.000000      0.000000      0.000000   \n",
       "24525     32.945114     43.001253     28.872605     22.747973   \n",
       "24526      0.772236      0.433510      1.478210      1.367791   \n",
       "24527      0.000000      0.000000      0.000000      0.000000   \n",
       "24528     16.502977     12.110023     14.354768     17.718008   \n",
       "24529      1.077352      0.000000      0.000000      0.000000   \n",
       "24530      8.207644      9.767945      5.746016      7.938302   \n",
       "24531   7032.415311   6123.059355   5507.448800   5569.149479   \n",
       "24532    138.819206    123.476630     94.945869    101.415155   \n",
       "24533      0.000000      0.971552      0.000000      0.000000   \n",
       "24534      9.290016      0.000000      2.425039      0.000000   \n",
       "24535     35.466212     41.974285      4.812858     29.691841   \n",
       "24536     79.283693    109.467617    104.728122    107.239231   \n",
       "24537      5.764022      0.861763      3.102230      2.426199   \n",
       "24538      0.000000      0.000000      0.000000      0.000000   \n",
       "24539      0.000000      0.000000      0.000000      0.000000   \n",
       "24540      0.000000      0.000000      0.000000      0.000000   \n",
       "24541      0.000000      0.000000      1.512578      0.000000   \n",
       "24542     12.679435     22.650132     21.824424      3.148545   \n",
       "24543     39.113089     38.935980     36.675994     37.184680   \n",
       "24544      9.815328      8.662501      9.859112      7.926904   \n",
       "24545      0.000000      0.000000      0.000000      0.000000   \n",
       "24546   4012.867833   3893.817815   3561.852333   3615.970743   \n",
       "24547      0.952927      2.144938      0.000000      1.112928   \n",
       "24548      0.000000      0.000000      0.000000      0.000000   \n",
       "24549      0.000000      0.000000      0.000000      0.000000   \n",
       "24550      0.000000      0.000000      0.000000      0.000000   \n",
       "\n",
       "                  gene_id     gene_name                   stop_codon  \n",
       "0      ENSG00000132932.17        ATP8A2    chr13 + 26019983 26019985  \n",
       "1      ENSG00000080815.19         PSEN1    chr14 + 73211856 73211858  \n",
       "2      ENSG00000109184.15       DCUN1D4     chr4 + 51848252 51848254  \n",
       "3      ENSG00000138395.17         CDK15   chr2 + 201807895 201807897  \n",
       "4      ENSG00000174720.16         LARP7   chr4 + 112646387 112646389  \n",
       "5      ENSG00000170100.13        ZNF778    chr16 + 89224743 89224745  \n",
       "6       ENSG00000130748.7       TMEM160    chr19 - 47045987 47045989  \n",
       "7      ENSG00000136933.17        RABEPK   chr9 + 125220566 125220568  \n",
       "8      ENSG00000164897.14         TMUB1   chr7 - 151081202 151081204  \n",
       "9      ENSG00000147536.12         GINS4     chr8 + 41537297 41537299  \n",
       "10      ENSG00000104980.8        TIMM44      chr19 - 7927187 7927189  \n",
       "11     ENSG00000133195.11      SLC39A11    chr17 - 72838038 72838040  \n",
       "12     ENSG00000143437.21          ARNT   chr1 - 150818022 150818024  \n",
       "13     ENSG00000213585.11         VDAC1   chr5 - 133972771 133972773  \n",
       "14     ENSG00000143252.15          SDHC   chr1 + 161339584 161339586  \n",
       "15     ENSG00000109787.13          KLF3     chr4 + 38697261 38697263  \n",
       "16     ENSG00000131951.11         LRRC9    chr14 + 60063360 60063362  \n",
       "17     ENSG00000198556.14        ZNF789     chr7 + 99485318 99485320  \n",
       "18     ENSG00000146592.17         CREB5     chr7 + 28819277 28819279  \n",
       "19     ENSG00000055609.18         KMT2C   chr7 - 152138784 152138786  \n",
       "20     ENSG00000198231.13         DDX42    chr17 + 63818396 63818398  \n",
       "21     ENSG00000071127.17          WDR1     chr4 - 10075378 10075380  \n",
       "22     ENSG00000172943.20          PHF8     chrX - 53939158 53939160  \n",
       "23     ENSG00000163681.16         SLMAP     chr3 + 57883981 57883983  \n",
       "24     ENSG00000135297.16          MTO1     chr6 + 73500585 73500587  \n",
       "25     ENSG00000204316.13        MRPL38    chr17 - 75898850 75898852  \n",
       "26     ENSG00000113522.14         RAD50   chr5 + 132557466 132557468  \n",
       "27     ENSG00000138031.14         ADCY3     chr2 - 24819932 24819934  \n",
       "28      ENSG00000125352.6       RNF113A   chrX - 119870582 119870584  \n",
       "29     ENSG00000153933.10          DGKE    chr17 + 56862264 56862266  \n",
       "...                   ...           ...                          ...  \n",
       "24521   ENSG00000269405.7          NXF2   chrX + 102326512 102326514  \n",
       "24522   ENSG00000285920.2    AC087721.2    chr15 + 41380184 41380186  \n",
       "24523   ENSG00000236362.8       GAGE12F     chrX + 49568112 49568114  \n",
       "24524   ENSG00000261832.6    AC138894.1    chr16 - 28479730 28479732  \n",
       "24525  ENSG00000185164.14         NOMO2    chr16 - 18500436 18500438  \n",
       "24526  ENSG00000183474.15       GTF2H2C     chr5 + 69566986 69566988  \n",
       "24527  ENSG00000159110.20        IFNAR2    chr21 + 33263498 33263500  \n",
       "24528   ENSG00000258790.1    AL121594.1    chr14 + 35273564 35273566  \n",
       "24529   ENSG00000261459.1    AC002310.4    chr16 - 30532499 30532501  \n",
       "24530  ENSG00000165060.14           FXN     chr9 + 69064948 69064950  \n",
       "24531   ENSG00000197238.5         H4C11     chr6 + 27824434 27824436  \n",
       "24532   ENSG00000177144.7        NUDT4B   chr1 + 148749496 148749498  \n",
       "24533  ENSG00000183054.11         RGPD6   chr2 - 110515611 110515613  \n",
       "24534  ENSG00000179304.16       FAM156B     chrX + 52907865 52907867  \n",
       "24535  ENSG00000106588.11         PSMA2     chr7 - 42917574 42917576  \n",
       "24536   ENSG00000267978.5       MAGEA9B   chrX - 149582232 149582234  \n",
       "24537   ENSG00000266086.3    AC015813.2    chr17 - 58005747 58005749  \n",
       "24538   ENSG00000288558.1  DUS4L-BCAP29   chr7 + 107575220 107575222  \n",
       "24539  ENSG00000256977.12         LIMS3   chr2 + 109919553 109919555  \n",
       "24540   ENSG00000286070.1          GGT1    chr22 + 24628837 24628839  \n",
       "24541  ENSG00000172058.15        SERF1A     chr5 + 70907804 70907806  \n",
       "24542  ENSG00000198064.13       NPIPB13    chr16 - 30240213 30240215  \n",
       "24543  ENSG00000166803.13         PCLAF    chr15 - 64366030 64366032  \n",
       "24544  ENSG00000092098.17         RNF31    chr14 + 24160571 24160573  \n",
       "24545  ENSG00000213648.11       SULT1A4    chr16 + 29464573 29464575  \n",
       "24546   ENSG00000272196.2        H2AC19   chr1 + 149851504 149851506  \n",
       "24547   ENSG00000170276.6         HSPB2  chr11 + 111913893 111913895  \n",
       "24548   ENSG00000272514.6       CFAP206     chr6 + 87464248 87464250  \n",
       "24549   ENSG00000269307.1    AC010463.1    chr19 + 17273943 17273945  \n",
       "24550   ENSG00000262304.2    AC027796.3      chr17 - 3610560 3610562  \n",
       "\n",
       "[24551 rows x 56 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_read_TPM_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:39.824584Z",
     "start_time": "2020-10-20T02:35:39.811105Z"
    }
   },
   "outputs": [],
   "source": [
    "def format_for_prism_normed(subset_df, output_prefix):\n",
    "    treatments = ['DMSO', 'SMG1i']\n",
    "    genotypes = ['WT', 'eIF4E2_4', 'GIGYF2_4']\n",
    "    replicates = ['1', '2']\n",
    "    molecules = ['RNA', 'mono', 'di']\n",
    "    headers = []\n",
    "    for treatment in treatments:\n",
    "        for genotype in genotypes:\n",
    "            for replicate in replicates:\n",
    "                headers.append('%s_%s_%s' % (genotype, treatment, replicate)) \n",
    "    \n",
    "    for molecule in molecules:\n",
    "        out_file=open('%s_%s.tsv' % (output_prefix, molecule), 'w')\n",
    "        out_file.write('%s\\t%s\\n' % ('tx_id', '\\t'.join(headers)))\n",
    "        for tx_id in subset_df['tx_id'].unique():\n",
    "            single_tx =  subset_df[subset_df['tx_id'] == tx_id]\n",
    "            NT_average = (single_tx['WT_DMSO_1_%s' % (molecule)].tolist()[0]+single_tx['WT_DMSO_2_%s' % (molecule)].tolist()[0])/2.0\n",
    "            data = []\n",
    "            for header in headers:\n",
    "                if '%s_%s' % (header, molecule) in single_tx.columns:\n",
    "                    if NT_average == 0:\n",
    "                        data.append(float('inf'))\n",
    "                    else:\n",
    "                        data.append(single_tx['%s_%s' % (header, molecule)].tolist()[0]/NT_average)\n",
    "            out_file.write('%s\\t%s\\n' % (tx_id, '\\t'.join(['%f' % datum for datum in data])))        \n",
    "        out_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:39.840575Z",
     "start_time": "2020-10-20T02:35:39.828291Z"
    }
   },
   "outputs": [],
   "source": [
    "def format_for_prism_counts(subset_df, output_prefix):\n",
    "    treatments = ['DMSO', 'SMG1i']\n",
    "    genotypes = ['WT', 'eIF4E2_4', 'GIGYF2_4']\n",
    "    replicates = ['1', '2']\n",
    "    molecules = ['RNA', 'mono', 'di']\n",
    "    headers = []\n",
    "    for treatment in treatments:\n",
    "        for genotype in genotypes:\n",
    "            for replicate in replicates:\n",
    "                headers.append('%s_%s_%s' % (genotype, treatment, replicate)) \n",
    "    \n",
    "    for molecule in molecules:\n",
    "        out_file=open('%s_%s.tsv' % (output_prefix, molecule), 'w')\n",
    "        out_file.write('%s\\t%s\\n' % ('tx_id', '\\t'.join(headers)))\n",
    "        for tx_id in subset_df['tx_id'].unique():\n",
    "            single_tx =  subset_df[subset_df['tx_id'] == tx_id]\n",
    "            data = []\n",
    "            for header in headers:\n",
    "                if '%s_%s' % (header, molecule) in single_tx.columns:\n",
    "                    data.append(single_tx['%s_%s' % (header, molecule)].tolist()[0])\n",
    "            out_file.write('%s\\t%s\\n' % (tx_id, '\\t'.join(['%f' % datum for datum in data])))        \n",
    "        out_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-24T19:04:02.109553Z",
     "start_time": "2020-09-24T19:04:01.946643Z"
    }
   },
   "outputs": [],
   "source": [
    "for gene_name in ['SRSF3', 'SRSF9', 'SRSF10', 'SRSF5', 'DDX6', 'SRSF7', 'SRSF6', 'SRSF4', 'SRSF1', 'SRSF2']:\n",
    "    format_for_prism_normed(combined_read_TPM_df[combined_read_TPM_df['gene_name']==gene_name], 'tpms_for_prism/20200924_%s_TPMs' % gene_name)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-20T02:35:40.051297Z",
     "start_time": "2020-10-20T02:35:39.843152Z"
    }
   },
   "outputs": [],
   "source": [
    "for gene_name in ['TPP2','FANCA','UBC','TUBA1C','ARIH1','MRPL37','SON','SNRNP70','RPLP0','UGAC_reporter','SDF2L1','H1-2','HNRNPH1','UAP1','MTRR','ALMS1']:\n",
    "    format_for_prism_normed(combined_read_TPM_df[combined_read_TPM_df['gene_name']==gene_name], 'tpms_for_prism/20201019_%s_TPMs' % gene_name)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-10-11T03:49:01.573952Z",
     "start_time": "2020-10-11T03:49:01.512856Z"
    }
   },
   "outputs": [],
   "source": [
    "for gene_name in ['UPF1', 'SMG7']:\n",
    "    format_for_prism_normed(combined_read_TPM_df[combined_read_TPM_df['gene_name']==gene_name], 'tpms_for_prism/20201010_%s_TPMs' % gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-24T20:02:14.688389Z",
     "start_time": "2020-09-24T20:02:14.525438Z"
    }
   },
   "outputs": [],
   "source": [
    "for gene_name in ['SRSF3', 'SRSF9', 'SRSF10', 'SRSF5', 'DDX6', 'SRSF7', 'SRSF6', 'SRSF4', 'SRSF1', 'SRSF2']:\n",
    "    format_for_prism_counts(combined_read_TPM_df[combined_read_TPM_df['gene_name']==gene_name], 'tpms_for_prism/20200924_%s_raw_TPMs' % gene_name)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-16T15:34:34.122879Z",
     "start_time": "2020-09-16T15:34:34.076732Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['WT_DMSO_1', 'WT_DMSO_2', 'eIF4E2_4_DMSO_1', 'eIF4E2_4_DMSO_2', 'GIGYF2_4_DMSO_1', 'GIGYF2_4_DMSO_2', 'WT_SMG1i_1', 'WT_SMG1i_2', 'eIF4E2_4_SMG1i_1', 'eIF4E2_4_SMG1i_2', 'GIGYF2_4_SMG1i_1', 'GIGYF2_4_SMG1i_2', 'NT1_1', 'NT1_2', 'SMG1i_1', 'SMG1i_2', 'eIF4E2_1', 'eIF4E2_2', 'GIGYF2_1', 'GIGYF2_2']\n",
      "ENST00000651723.1\n",
      "ENST00000652016.1\n",
      "ENST00000533482.5\n",
      "ENST00000378823.8\n",
      "ENST00000651160.1\n"
     ]
    }
   ],
   "source": [
    "format_for_prism(combined_read_TPM_df[combined_read_TPM_df['gene_name']=='RAD50'], 'tpms_for_prism/20200823_RAD50_TPMs')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Look at correlation between TPMs - for individual transcripts, grouped by gene, and grouped by stop codon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-23T22:08:56.491107Z",
     "start_time": "2020-08-23T22:06:52.260Z"
    }
   },
   "outputs": [],
   "source": [
    "combined_read_counts_df.columns[1:-3].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-23T22:08:56.493035Z",
     "start_time": "2020-08-23T22:06:53.673Z"
    }
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "from adjustText import adjust_text\n",
    "import math\n",
    "\n",
    "plt.rcParams['pdf.fonttype'] = 42 #leaves most text as actual text in PDFs, not outlines\n",
    "##plot CDF or mRNA RPKMs\n",
    "black = (0,0,0)\n",
    "gray = (0.5,0.5,0.5)\n",
    "orange = (230/255.0,159/255.0,0)\n",
    "skyBlue = (86/255.0,180/255.0,233/255.0)\n",
    "bluishGreen = (0,158/255.0,115/255.0)\n",
    "yellow = (240/255.0,228/255.0,66/255.0)\n",
    "blue = (0,114/255.0,178/255.0)\n",
    "vermillion = (213/255.0,94/255.0,0)\n",
    "reddishPurple = (204/255.0,121/255.0,167/255.0)\n",
    "colors = [black, orange, skyBlue, bluishGreen, vermillion, blue, reddishPurple, yellow]\n",
    "rainbow = [black, vermillion, orange, bluishGreen, blue, reddishPurple, 'violet']\n",
    "\n",
    "treatments = ['DMSO', 'SMG1i']\n",
    "genotypes = ['WT', 'eIF4E2_4', 'GIGYF2_4']\n",
    "replicates = ['1', '2']\n",
    "molecules = ['RNA', 'mono', 'di']\n",
    "\n",
    "df = combined_read_TPM_df\n",
    "\n",
    "for molecule in molecules:\n",
    "    datasets = []\n",
    "    for treatment in treatments:\n",
    "            for genotype in genotypes:\n",
    "                for replicate in replicates:\n",
    "                    datasets.append('%s_%s_%s_%s' % (genotype, treatment, replicate, molecule))\n",
    "    if molecule != 'di':\n",
    "        for genotype in ['NT1', 'SMG1i', 'eIF4E2', 'GIGYF2']:\n",
    "            for replicate in replicates:\n",
    "                datasets.append('%s_%s_%s' % (genotype, replicate, molecule))   \n",
    "\n",
    "    num_plots_wide = len(datasets)\n",
    "    num_plots_high = num_plots_wide\n",
    "    fig = plt.figure(figsize=(2*num_plots_wide, 2*num_plots_high))\n",
    "    plot_index = 1\n",
    "\n",
    "    for i in range(len(datasets)):\n",
    "        for j in range(len(datasets)):\n",
    "            plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "            if j < i:\n",
    "                df.plot.scatter(datasets[j], datasets[i], color=black, s=1, alpha=1, ax=plot)\n",
    "                plot.set_xlabel(datasets[j])\n",
    "                plot.set_ylabel(datasets[i])\n",
    "                plot.set_xscale('log')\n",
    "                plot.set_yscale('log')\n",
    "                plot.set_ylim(10**-4, 10**6)\n",
    "                plot.set_xlim(10**-4, 10**6)\n",
    "            elif j == i:\n",
    "                #no point comparing same dataset\n",
    "                plot.spines['left'].set_visible(False)\n",
    "                plot.spines['bottom'].set_visible(False)\n",
    "                plot.set_xticks([])\n",
    "                plot.set_yticks([])\n",
    "            else:\n",
    "                #plot the correlations up top\n",
    "                subset_df = pd.DataFrame()\n",
    "                subset_df[datasets[i]] = df[datasets[i]].apply(log_trans)\n",
    "                subset_df[datasets[j]] = df[datasets[j]].apply(log_trans)\n",
    "                subset_df=subset_df.dropna()\n",
    "                pr, pp = stats.pearsonr(subset_df[datasets[j]], subset_df[datasets[i]])\n",
    "                sr, sp = stats.pearsonr(subset_df[datasets[j]], subset_df[datasets[i]])\n",
    "                plot.set_xlabel(datasets[j])\n",
    "                plot.set_ylabel(datasets[i])\n",
    "                plot.text(-4.5, -2, 'r=%.3f\\nrho=%.3f\\n' % (pr, sr), fontsize=16)\n",
    "                plot.set_ylim(-5, 5)\n",
    "                plot.set_xlim(-5, 5)\n",
    "                plot.set_xticks([])\n",
    "                plot.set_yticks([])\n",
    "            plot.spines['right'].set_visible(False)\n",
    "            plot.spines['top'].set_visible(False)\n",
    "            plot.set_aspect('equal')\n",
    "            plot_index+=1\n",
    "\n",
    "    #lg=plt.legend(loc=4,prop={'size':12}, labelspacing=0.2)\n",
    "    #lg.draw_frame(False)\n",
    "    plt.tight_layout()\n",
    "    plt.savefig('20200823_all_tx_%s_TPM_corr.png' % (molecule), transparent='True', format='png', dpi = 400)\n",
    "    #plt.savefig('transcript_TPM_corr.pdf', transparent='True', format='pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-23T22:08:56.494455Z",
     "start_time": "2020-08-23T22:06:57.955Z"
    }
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "from adjustText import adjust_text\n",
    "import math\n",
    "\n",
    "plt.rcParams['pdf.fonttype'] = 42 #leaves most text as actual text in PDFs, not outlines\n",
    "##plot CDF or mRNA RPKMs\n",
    "black = (0,0,0)\n",
    "gray = (0.5,0.5,0.5)\n",
    "orange = (230/255.0,159/255.0,0)\n",
    "skyBlue = (86/255.0,180/255.0,233/255.0)\n",
    "bluishGreen = (0,158/255.0,115/255.0)\n",
    "yellow = (240/255.0,228/255.0,66/255.0)\n",
    "blue = (0,114/255.0,178/255.0)\n",
    "vermillion = (213/255.0,94/255.0,0)\n",
    "reddishPurple = (204/255.0,121/255.0,167/255.0)\n",
    "colors = [black, orange, skyBlue, bluishGreen, vermillion, blue, reddishPurple, yellow]\n",
    "rainbow = [black, vermillion, orange, bluishGreen, blue, reddishPurple, 'violet']\n",
    "\n",
    "treatments = ['DMSO', 'SMG1i']\n",
    "genotypes = ['WT', 'eIF4E2_4', 'GIGYF2_4']\n",
    "replicates = ['1', '2']\n",
    "molecules = ['RNA', 'mono', 'di']\n",
    "\n",
    "df = tpm_by_gene_df\n",
    "\n",
    "for molecule in molecules:\n",
    "    datasets = []\n",
    "    for treatment in treatments:\n",
    "        for genotype in genotypes:\n",
    "            for replicate in replicates:\n",
    "                datasets.append('%s_%s_%s_%s' % (genotype, treatment, replicate, molecule))\n",
    "    \n",
    "    num_plots_wide = len(datasets)\n",
    "    num_plots_high = num_plots_wide\n",
    "    fig = plt.figure(figsize=(2*num_plots_wide, 2*num_plots_high))\n",
    "    plot_index = 1\n",
    "\n",
    "    for i in range(len(datasets)):\n",
    "        for j in range(len(datasets)):\n",
    "            plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "            if j < i:\n",
    "                df.plot.scatter(datasets[j], datasets[i], color=black, s=1, alpha=1, ax=plot)\n",
    "                plot.set_xlabel(datasets[j])\n",
    "                plot.set_ylabel(datasets[i])\n",
    "                plot.set_xscale('log')\n",
    "                plot.set_yscale('log')\n",
    "                plot.set_ylim(10**-4, 10**6)\n",
    "                plot.set_xlim(10**-4, 10**6)\n",
    "            elif j == i:\n",
    "                #no point comparing same dataset\n",
    "                plot.spines['left'].set_visible(False)\n",
    "                plot.spines['bottom'].set_visible(False)\n",
    "                plot.set_xticks([])\n",
    "                plot.set_yticks([])\n",
    "            else:\n",
    "                #plot the correlations up top\n",
    "                subset_df = pd.DataFrame()\n",
    "                subset_df[datasets[i]] = df[datasets[i]].apply(log_trans)\n",
    "                subset_df[datasets[j]] = df[datasets[j]].apply(log_trans)\n",
    "                subset_df=subset_df.dropna()\n",
    "                pr, pp = stats.pearsonr(subset_df[datasets[j]], subset_df[datasets[i]])\n",
    "                sr, sp = stats.pearsonr(subset_df[datasets[j]], subset_df[datasets[i]])\n",
    "                plot.set_xlabel(datasets[j])\n",
    "                plot.set_ylabel(datasets[i])\n",
    "                plot.text(-4.5, -2, 'r=%.3f\\nrho=%.3f\\n' % (pr, sr), fontsize=16)\n",
    "                plot.set_ylim(-5, 5)\n",
    "                plot.set_xlim(-5, 5)\n",
    "                plot.set_xticks([])\n",
    "                plot.set_yticks([])\n",
    "            plot.spines['right'].set_visible(False)\n",
    "            plot.spines['top'].set_visible(False)\n",
    "            plot.set_aspect('equal')\n",
    "            plot_index+=1\n",
    "\n",
    "    #lg=plt.legend(loc=4,prop={'size':12}, labelspacing=0.2)\n",
    "    #lg.draw_frame(False)\n",
    "    plt.tight_layout()\n",
    "    plt.savefig('20200823_by_gene_%s_TPM_corr.png' % (molecule), transparent='True', format='png', dpi = 400)\n",
    "    #plt.savefig('transcript_TPM_corr.pdf', transparent='True', format='pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-24T17:11:11.846512Z",
     "start_time": "2020-07-24T17:09:32.912304Z"
    }
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "from adjustText import adjust_text\n",
    "import math\n",
    "\n",
    "plt.rcParams['pdf.fonttype'] = 42 #leaves most text as actual text in PDFs, not outlines\n",
    "##plot CDF or mRNA RPKMs\n",
    "black = (0,0,0)\n",
    "gray = (0.5,0.5,0.5)\n",
    "orange = (230/255.0,159/255.0,0)\n",
    "skyBlue = (86/255.0,180/255.0,233/255.0)\n",
    "bluishGreen = (0,158/255.0,115/255.0)\n",
    "yellow = (240/255.0,228/255.0,66/255.0)\n",
    "blue = (0,114/255.0,178/255.0)\n",
    "vermillion = (213/255.0,94/255.0,0)\n",
    "reddishPurple = (204/255.0,121/255.0,167/255.0)\n",
    "colors = [black, orange, skyBlue, bluishGreen, vermillion, blue, reddishPurple, yellow]\n",
    "rainbow = [black, vermillion, orange, bluishGreen, blue, reddishPurple, 'violet']\n",
    "\n",
    "treatments = ['DMSO', 'SMG1i']\n",
    "genotypes = ['WT', 'eIF4E2_4', 'GIGYF2_4']\n",
    "replicates = ['1', '2']\n",
    "molecules = ['RNA', 'mono', 'di']\n",
    "\n",
    "df = TPM_by_stop_df\n",
    "\n",
    "for molecule in molecules:\n",
    "    datasets = []\n",
    "    for treatment in treatments:\n",
    "        for genotype in genotypes:\n",
    "            for replicate in replicates:\n",
    "                datasets.append('%s_%s_%s_%s' % (genotype, treatment, replicate, molecule))\n",
    "    \n",
    "    num_plots_wide = len(datasets)\n",
    "    num_plots_high = num_plots_wide\n",
    "    fig = plt.figure(figsize=(2*num_plots_wide, 2*num_plots_high))\n",
    "    plot_index = 1\n",
    "\n",
    "    for i in range(len(datasets)):\n",
    "        for j in range(len(datasets)):\n",
    "            plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "            if j < i:\n",
    "                df.plot.scatter(datasets[j], datasets[i], color=black, s=1, alpha=1, ax=plot)\n",
    "                plot.set_xlabel(datasets[j])\n",
    "                plot.set_ylabel(datasets[i])\n",
    "                plot.set_xscale('log')\n",
    "                plot.set_yscale('log')\n",
    "                plot.set_ylim(10**-4, 10**6)\n",
    "                plot.set_xlim(10**-4, 10**6)\n",
    "            elif j == i:\n",
    "                #no point comparing same dataset\n",
    "                plot.spines['left'].set_visible(False)\n",
    "                plot.spines['bottom'].set_visible(False)\n",
    "                plot.set_xticks([])\n",
    "                plot.set_yticks([])\n",
    "            else:\n",
    "                #plot the correlations up top\n",
    "                subset_df = pd.DataFrame()\n",
    "                subset_df[datasets[i]] = df[datasets[i]].apply(log_trans)\n",
    "                subset_df[datasets[j]] = df[datasets[j]].apply(log_trans)\n",
    "                subset_df=subset_df.dropna()\n",
    "                pr, pp = stats.pearsonr(subset_df[datasets[j]], subset_df[datasets[i]])\n",
    "                sr, sp = stats.pearsonr(subset_df[datasets[j]], subset_df[datasets[i]])\n",
    "                plot.set_xlabel(datasets[j])\n",
    "                plot.set_ylabel(datasets[i])\n",
    "                plot.text(-4.5, -2, 'r=%.3f\\nrho=%.3f\\n' % (pr, sr), fontsize=16)\n",
    "                plot.set_ylim(-5, 5)\n",
    "                plot.set_xlim(-5, 5)\n",
    "                plot.set_xticks([])\n",
    "                plot.set_yticks([])\n",
    "            plot.spines['right'].set_visible(False)\n",
    "            plot.spines['top'].set_visible(False)\n",
    "            plot.set_aspect('equal')\n",
    "            plot_index+=1\n",
    "\n",
    "    #lg=plt.legend(loc=4,prop={'size':12}, labelspacing=0.2)\n",
    "    #lg.draw_frame(False)\n",
    "    plt.tight_layout()\n",
    "    plt.savefig('20200723_by_stop_%s_TPM_corr.png' % (molecule), transparent='True', format='png', dpi = 400)\n",
    "    #plt.savefig('transcript_TPM_corr.png', transparent='True', format='pdf')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-19T16:26:13.858552Z",
     "start_time": "2020-07-19T16:26:13.839485Z"
    }
   },
   "source": [
    "## Code to check for transcripts without stop codons - not needed now?\n",
    "I think it is a problem that some of the genes have no annotated stop codon - I want to look at the extent of the problem here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-21T13:45:05.504271Z",
     "start_time": "2020-07-21T13:45:05.376712Z"
    },
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "#set up plotting\n",
    "%matplotlib inline\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "plt.rcParams['pdf.fonttype'] = 42 #leaves most text as actual text in PDFs, not outlines\n",
    "# Say, \"the default sans-serif font is arial\"\n",
    "mpl.rcParams['font.sans-serif'] = \"Arial\"\n",
    "# Then, \"ALWAYS use sans-serif fonts\"\n",
    "mpl.rcParams['font.family'] = \"sans-serif\"\n",
    "#colorblind safe colors\n",
    "black = (0,0,0)\n",
    "gray = (0.5,0.5,0.5)\n",
    "orange = (230/255.0,159/255.0,0)\n",
    "skyBlue = (86/255.0,180/255.0,233/255.0)\n",
    "bluishGreen = (0,158/255.0,115/255.0)\n",
    "yellow = (240/255.0,228/255.0,66/255.0)\n",
    "blue = (0,114/255.0,178/255.0)\n",
    "vermillion = (213/255.0,94/255.0,0)\n",
    "reddishPurple = (204/255.0,121/255.0,167/255.0)\n",
    "colors = [gray, black, blue, orange, bluishGreen, skyBlue, vermillion, reddishPurple, yellow, 'red', 'green']\n",
    "rainbow = [black, vermillion, orange, bluishGreen, blue, reddishPurple, 'violet']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-23T21:37:14.061626Z",
     "start_time": "2020-07-23T21:37:06.691851Z"
    }
   },
   "outputs": [],
   "source": [
    "#histograms of number and fraction of transcripts without an annotated stop codon\n",
    "num_unannotated_hist = []\n",
    "frac_unannotated_hist = []\n",
    "\n",
    "grouped_counts = combined_read_counts_df.groupby('gene_id')\n",
    "for gene_id in combined_read_counts_df['gene_id'].unique():\n",
    "    gene_transcript_data = grouped_counts.get_group(gene_id)\n",
    "    tx_ids = gene_transcript_data['tx_id'].unique()\n",
    "    num_unannotated_stops = len([tx_id for tx_id in tx_ids if get_genomic_stop_coords(tx_id)=='not annotated'])\n",
    "    frac_unannotated_stops = float(num_unannotated_stops)/float(len(tx_ids))\n",
    "    num_unannotated_hist.append(num_unannotated_stops)\n",
    "    frac_unannotated_hist.append(frac_unannotated_stops)\n",
    "    \n",
    "plots = []\n",
    "num_plots_wide = 2\n",
    "num_plots_high = 3\n",
    "fig = plt.figure(figsize=(4*num_plots_wide, 4*num_plots_high))\n",
    "\n",
    "#firs plot number of tx w/ no stop codon per gene\n",
    "plot_index = 1\n",
    "plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "plot.hist(num_unannotated_hist, color=ribo_utils.black, lw=2, align='mid', bins = range(max(num_unannotated_hist)))\n",
    "plot.set_ylabel(\"# genes\")\n",
    "plot.set_xlabel(\"# transcripts with unannotated stop\")\n",
    "plots.append(plot)\n",
    "\n",
    "plot_index += 1\n",
    "plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "plot.hist(frac_unannotated_hist, color=ribo_utils.black, lw=2, align='mid', bins = np.arange(0, max(frac_unannotated_hist), 0.05))\n",
    "plot.set_ylabel(\"# genes\")\n",
    "plot.set_xlabel(\"fraction of transcripts with unannotated stop\")\n",
    "plots.append(plot)\n",
    "\n",
    "plot_index += 1\n",
    "plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "plot.hist(num_unannotated_hist, color=ribo_utils.black, lw=2, align='mid', bins = range(max(num_unannotated_hist)+1))\n",
    "plot.set_ylabel(\"# genes\")\n",
    "plot.set_xlabel(\"# transcripts with unannotated stop\")\n",
    "plot.set_ylim(0, 1000)\n",
    "plots.append(plot)\n",
    "\n",
    "plot_index += 1\n",
    "plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "plot.hist(frac_unannotated_hist, color=ribo_utils.black, lw=2, align='mid', bins = np.arange(0, max(frac_unannotated_hist)+0.05, 0.05))\n",
    "plot.set_ylabel(\"# genes\")\n",
    "plot.set_xlabel(\"fraction of transcripts with unannotated stop\")\n",
    "plot.set_ylim(0, 1000)\n",
    "plots.append(plot)\n",
    "\n",
    "plot_index += 1\n",
    "plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "plot.hist(num_unannotated_hist, color=ribo_utils.black, lw=2, align='mid', normed=True, cumulative=True, bins = 1000, histtype='step')\n",
    "plot.set_ylabel(\"frac of genes\")\n",
    "plot.set_xlabel(\"# transcripts with unannotated stop\")\n",
    "#plot.set_ylim(0, 1000)\n",
    "plots.append(plot)\n",
    "\n",
    "plot_index += 1\n",
    "plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "plot.hist(frac_unannotated_hist, color=ribo_utils.black, lw=2, align='mid', normed=True, cumulative=True, bins = 1000, histtype='step')\n",
    "plot.set_ylabel(\"frac of genes\")\n",
    "plot.set_xlabel(\"fraction of transcripts with unannotated stop\")\n",
    "#plot.set_ylim(0, 1)\n",
    "plots.append(plot)\n",
    "\n",
    "for plot in plots:\n",
    "    plot.spines['right'].set_visible(False)\n",
    "    plot.spines['top'].set_visible(False)\n",
    "    #plot.set_xticks(range(11))\n",
    "    #plot.set_ylim(0, 1)\n",
    "    #plot.set_xlim(0.5, 10)\n",
    "\n",
    "#lg=plt.legend(prop={'size':12}, labelspacing=0.2, bbox_to_anchor=(1.04,1), loc=\"upper left\")\n",
    "#lg.draw_frame(False)\n",
    "#plt.tight_layout()\n",
    "plt.savefig('E8.019_20200721_transcripts_without_stop.pdf', transparent='True', format='pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-19T17:49:02.262249Z",
     "start_time": "2020-07-19T17:49:02.250048Z"
    }
   },
   "outputs": [],
   "source": [
    "datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-21T13:53:33.312761Z",
     "start_time": "2020-07-21T13:46:03.860541Z"
    }
   },
   "outputs": [],
   "source": [
    "#fraction of reads for each gene mapping to transcripts without an annotated stop codon\n",
    "from collections import defaultdict\n",
    "datasets = list(combined_read_counts_df.columns[1:-3])\n",
    "print datasets\n",
    "fractions_mapping_to_unannotated = defaultdict(list)\n",
    "number_mapping_to_unannotated = defaultdict(float)\n",
    "total_mapping = defaultdict(float)\n",
    "\n",
    "\n",
    "gene_grouped_counts = combined_read_counts_df.groupby('gene_id')\n",
    "tx_grouped_counts = combined_read_counts_df.groupby('tx_id')\n",
    "\n",
    "for gene_id in combined_read_counts_df['gene_id'].unique():\n",
    "    gene_transcript_data = grouped_counts.get_group(gene_id)\n",
    "    tx_ids = gene_transcript_data['tx_id'].unique()\n",
    "    for dataset in datasets:\n",
    "        total_gene_mapping = 0.\n",
    "        unannotated_mapping = 0.\n",
    "        for tx_id in tx_ids:\n",
    "            tx_data = tx_grouped_counts.get_group(tx_id)\n",
    "            #print(tx_data[dataset].tolist()[0])\n",
    "            counts = float(tx_data[dataset].tolist()[0])\n",
    "            total_gene_mapping += counts\n",
    "            if get_genomic_stop_coords(tx_id)=='not annotated':\n",
    "                unannotated_mapping += counts\n",
    "        if total_gene_mapping > 0:\n",
    "            total_mapping[dataset] += total_gene_mapping\n",
    "            number_mapping_to_unannotated[dataset] += unannotated_mapping\n",
    "            fractions_mapping_to_unannotated[dataset].append(float(unannotated_mapping)/float(total_gene_mapping))\n",
    "\n",
    "for dataset in datasets:\n",
    "    print(dataset, 'fraction_mapping_no_stop:', number_mapping_to_unannotated[dataset]/total_mapping[dataset])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-07-21T13:53:34.675778Z",
     "start_time": "2020-07-21T13:53:33.316458Z"
    }
   },
   "outputs": [],
   "source": [
    "dataset_groups = [datasets[:4], datasets[12:16],datasets[24:28]]\n",
    "plots = []\n",
    "num_plots_wide = 3\n",
    "num_plots_high = 3\n",
    "fig = plt.figure(figsize=(4*num_plots_wide, 4*num_plots_high))\n",
    "\n",
    "#firs plot number of tx w/ no stop codon per gene\n",
    "plot_index = 1\n",
    "for dataset_group in dataset_groups:\n",
    "    plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "    color_index = 0\n",
    "    for dataset in dataset_group:\n",
    "        plot.hist(fractions_mapping_to_unannotated[dataset], color=ribo_utils.colors[color_index], lw=2, histtype='step', align='mid', \n",
    "                  bins = np.arange(0,1.05,0.05), label = dataset)\n",
    "        color_index += 1\n",
    "    plot.set_ylabel(\"# genes\")\n",
    "    plot.set_xlabel(\"fraction reads mapping to no-stop tx\")\n",
    "    plot.set_title(dataset_group[0].split('_')[-1])\n",
    "    plots.append(plot)\n",
    "    plot_index += 1\n",
    "for dataset_group in dataset_groups:\n",
    "    plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "    color_index = 0\n",
    "    for dataset in dataset_group:\n",
    "        plot.hist(fractions_mapping_to_unannotated[dataset], color=ribo_utils.colors[color_index], lw=2, histtype='step', align='mid', \n",
    "                  bins = np.arange(0,1.05,0.05), label = dataset)\n",
    "        color_index += 1\n",
    "    plot.set_ylim(0, 300)\n",
    "    plot.set_ylabel(\"# genes\")\n",
    "    plot.set_xlabel(\"fraction reads mapping to no-stop tx\")\n",
    "    plot.set_title(dataset_group[0].split('_')[-1])\n",
    "    plots.append(plot)\n",
    "    plot_index += 1\n",
    "\n",
    "for dataset_group in dataset_groups:\n",
    "    plot = fig.add_subplot(num_plots_high, num_plots_wide, plot_index)\n",
    "    color_index = 0\n",
    "    for dataset in dataset_group:\n",
    "        plot.hist(fractions_mapping_to_unannotated[dataset], color=ribo_utils.colors[color_index], lw=2, normed=True, cumulative=True, histtype='step', align='mid', \n",
    "                  bins = 10000, label = dataset)\n",
    "        color_index += 1\n",
    "    plot.set_ylim(0.7, 1.01)\n",
    "    plot.set_ylabel(\"fraction genes\")\n",
    "    plot.set_xlabel(\"fraction reads mapping to no-stop tx\")\n",
    "    plot.set_title(dataset_group[0].split('_')[-1])\n",
    "    plots.append(plot)\n",
    "    plot_index += 1\n",
    "    \n",
    "for plot in plots:\n",
    "    plot.spines['right'].set_visible(False)\n",
    "    plot.spines['top'].set_visible(False)\n",
    "    #plot.set_xticks(range(11))\n",
    "    #plot.set_ylim(0, 1)\n",
    "    #plot.set_xlim(0.5, 10)\n",
    "\n",
    "#lg=plt.legend(prop={'size':12}, labelspacing=0.2, bbox_to_anchor=(1.04,1), loc=\"upper left\")\n",
    "#lg.draw_frame(False)\n",
    "plt.tight_layout()\n",
    "plt.savefig('E8.019_20200721_frac_reads_transcripts_without_stop.pdf', transparent='True', format='pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
