{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-18T22:05:09.496529Z",
     "start_time": "2020-11-18T22:05:09.453Z"
    }
   },
   "outputs": [],
   "source": [
    "#installation, may not be needed\n",
    "#if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "#    install.packages(\"BiocManager\")\n",
    "\n",
    "#BiocManager::install(\"DESeq2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-18T22:05:15.637871Z",
     "start_time": "2020-11-18T22:05:10.367Z"
    }
   },
   "outputs": [],
   "source": [
    "suppressMessages(library(DESeq2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-18T22:16:58.874664Z",
     "start_time": "2020-11-18T22:16:34.914Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#4A3 knockdown vs scramble HEPG2\n",
    "count_file <- \"../0_download_run_salmon/20201117_EJC_factors_stop_collapsed_counts.tsv\"\n",
    "row_names <- \"tx_id\"\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('HepG2_ctrl_1', 'HepG2_ctrl_2', 'HepG2_EIF4A3_1', 'HepG2_EIF4A3_2')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 2), rep(\"exp\", 2)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/HEPG2_EIF4A3.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#4A3 knockdown vs scramble K562\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('K562_ctrl_for_EIF4A3_1', 'K562_ctrl_for_EIF4A3_2', 'K562_EIF4A3_1', 'K562_EIF4A3_2')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 2), rep(\"exp\", 2)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/K562_EIF4A3.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#MAGOH knockdown vs scramble HEPG2\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('HepG2_ctrl_1', 'HepG2_ctrl_2', 'HepG2_MAGOH_1', 'HepG2_MAGOH_2')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 2), rep(\"exp\", 2)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/HEPG2_MAGOH.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)\n",
    "\n",
    "#MAGOH knockdown vs scramble K562\n",
    "countdata <- read.csv(count_file, row.names=row_names, sep='\\t')[ ,c('K562_ctrl_for_MAGOH_1', 'K562_ctrl_for_MAGOH_2', 'K562_MAGOH_1', 'K562_MAGOH_2')]\n",
    "countdata <- round(countdata)\n",
    "condition <- factor(c(rep(\"ctl\", 2), rep(\"exp\", 2)))\n",
    "# Create a coldata frame and instantiate the DESeqDataSet. See ?DESeqDataSetFromMatrix\n",
    "coldata <- data.frame(row.names=colnames(countdata), condition)\n",
    "dds <- DESeqDataSetFromMatrix(countData=countdata, colData=coldata, design=~condition)\n",
    "# Run the DESeq pipeline\n",
    "dds <- DESeq(dds)\n",
    "#filter for at least 30 counts between all datasets\n",
    "dds <- dds[ rowSums(counts(dds)) > 30, ]\n",
    "# Get differential expression results\n",
    "res <- results(dds)\n",
    "## Order by adjusted p-value\n",
    "res <- res[order(res$padj), ]\n",
    "## Merge with normalized count data\n",
    "#res <- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=\"row.names\", sort=FALSE)\n",
    "#names(res)[1] <- \"Transcript\"\n",
    "## Write results\n",
    "write.table(res, file='DESeq_results/K562_MAGOH.txt', quote=F, sep='\\t', col.names = NA, row.names = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
